{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "cdbb4d45",
   "metadata": {},
   "outputs": [],
   "source": [
    "# HMR Annotation - Hematopoietic Lineage"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b58d2738",
   "metadata": {},
   "source": [
    "# Format RefSeq promoters and exons"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "03f91225",
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "# Working directory: \n",
    "cd /data/hodges_lab/Tim/finalAnalyses_HMRs/hp_lineage/referenceFiles\n",
    "\n",
    "######  Get promoters\n",
    "# Starting files: ncbiRefSeqCurated_promotersUpstream2000.txt\n",
    "#### (1) Add ID column baed on the current ID column ($4) so that IDs match across files (e.g. NM_001308203.2_exon_0_0_chr1_66999276_f --> NM_001308203.2); truncate after second _\n",
    "paste ncbiRefSeqCurated_promotersUpstream2000.txt <(awk 'BEGIN{OFS=FS=\"\\t\"}{print $4}' ncbiRefSeqCurated_promotersUpstream2000.txt | awk 'BEGIN{OFS=\"_\";FS=\"_\"}{print $1,$2}' - ) > ncbiRefSeqCurated_promotersUpstream2000.IDcolAdded.txt \n",
    "#### (2) Filter out extras cols and rows that are not NM genes\n",
    "awk 'BEGIN{OFS=FS=\"\\t\"}{if ($7~/NM/) print $1,$2,$3,$7,$6}' ncbiRefSeqCurated_promotersUpstream2000.IDcolAdded.txt > ncbiRefSeqCurated_NM.promotersUpstream2000.txt # 59425\n",
    "# \t\t(2a) Sort with bash \n",
    "sort -t$'\\t' -k1,1 -k2,2n -k3,3n ncbiRefSeqCurated_NM.promotersUpstream2000.txt > ncbiRefSeqCurated_NM.promotersUpstream2000.bashSorted.txt # 59425\n",
    "\n",
    "\n",
    "#### (3) Check the strand direction, and add 1000 downstream to each promoter\n",
    "# Instead of writing a horrible awk script, might be clearer and safer to write two awk scripts, one for each strand\n",
    "# \t\t(3a) Add a few columns at the end  to check genome lengths \n",
    "# sort -k1,1n -k2,2n -k3,3n  hg19.chrom.sizes > hg19.chrom.sizes.sorted\n",
    "join -t$'\\t' -j 1 -o 1.1,1.2,1.3,1.4,1.5,2.2 <(sort -t$'\\t' -k1,1 -k2,2n -k3,3n ncbiRefSeqCurated_NM.promotersUpstream2000.bashSorted.txt) <(sort -t$'\\t' -k1,1 -k2,2n -k3,3n hg19.chrom.sizes.sorted) > ncbiRefSeqCurated_NM.promotersUpstream2000_withChromLengths.txt # 56099\n",
    "###\t \t(3a1) CHECK TO MAKE SURE THIS SHORTER FILE IS KOSHER\n",
    " #  59425 ncbiRefSeqCurated_NM.promotersUpstream2000.txt\n",
    " #  56099 ncbiRefSeqCurated_NM.promotersUpstream2000_withChromLengths.txt\n",
    "\n",
    "# WHY IS THE withChromLength file shorter! Find the lines lost\n",
    "comm -2 -3 <(sort ncbiRefSeqCurated_NM.promotersUpstream2000.bashSorted.txt) <(sort <(awk 'BEGIN{OFS=FS=\"\\t\"}{print $1,$2,$3,$4,$5}' ncbiRefSeqCurated_NM.promotersUpstream2000_withChromLengths.txt)) | sort -t$'\\t' -k1,1 -k2,2n -k3,3n - > PromotersWithoutAChromSizeRef.txt\n",
    "\n",
    "# Okay, check if any HMRs are on these contigs\n",
    "awk 'BEGIN{OFS=FS=\"\\t\"}{print $1}' /data/hodges_lab/Tim/finalAnalyses_HMRs/HMRs_goodGenes/startingBEDs/Adrenal.hmr.bed > Adrenal.hmr.bed.chrCol.txt\n",
    "awk 'BEGIN{OFS=FS=\"\\t\"}{print $1}' /data/hodges_lab/Tim/finalAnalyses_HMRs/HMRs_goodGenes/startingBEDs/Bcell.hmr.bed > Bcell.hmr.bed.chrCol.txt\n",
    "awk 'BEGIN{OFS=FS=\"\\t\"}{print $1}' /data/hodges_lab/Tim/finalAnalyses_HMRs/HMRs_goodGenes/startingBEDs/fHeart.hmr.bed > fHeart.hmr.bed.chrCol.txt\n",
    "awk 'BEGIN{OFS=FS=\"\\t\"}{print $1}' /data/hodges_lab/Tim/finalAnalyses_HMRs/HMRs_goodGenes/startingBEDs/fSpinal.hmr.bed > fSpinal.hmr.bed.chrCol.txt\n",
    "awk 'BEGIN{OFS=FS=\"\\t\"}{print $1}' /data/hodges_lab/Tim/finalAnalyses_HMRs/HMRs_goodGenes/startingBEDs/H1ESC.hmr.bed > H1ESC.hmr.bed.chrCol.txt\n",
    "awk 'BEGIN{OFS=FS=\"\\t\"}{print $1}' /data/hodges_lab/Tim/finalAnalyses_HMRs/HMRs_goodGenes/startingBEDs/Liver.hmr.bed > Liver.hmr.bed.chrCol.txt\n",
    "\n",
    "awk 'BEGIN{OFS=FS=\"\\t\"}{print $1}' PromotersWithoutAChromSizeRef.txt > PromotersWithoutAChromSizeRef.chrCol.txt\n",
    "comm -1 -2 <(sort Adrenal.hmr.bed.chrCol.txt) <(sort PromotersWithoutAChromSizeRef.chrCol.txt)\n",
    "comm -1 -2 <(sort Bcell.hmr.bed.chrCol.txt) <(sort PromotersWithoutAChromSizeRef.chrCol.txt)\n",
    "comm -1 -2 <(sort fHeart.hmr.bed.chrCol.txt) <(sort PromotersWithoutAChromSizeRef.chrCol.txt)\n",
    "comm -1 -2 <(sort fSpinal.hmr.bed.chrCol.txt) <(sort PromotersWithoutAChromSizeRef.chrCol.txt)\n",
    "comm -1 -2 <(sort H1ESC.hmr.bed.chrCol.txt) <(sort PromotersWithoutAChromSizeRef.chrCol.txt)\n",
    "comm -1 -2 <(sort Liver.hmr.bed.chrCol.txt) <(sort PromotersWithoutAChromSizeRef.chrCol.txt)\n",
    "# NONE ARE IN THE CONTIGS! Throw them out! Maybe. \n",
    "\n",
    "\n",
    "#  ADD DOWNSTREAM 1000\n",
    "###\t\t(3b) + strand; Add 1000, check if this passes the genome boundary, and if so, print that\n",
    "awk 'BEGIN{OFS=FS=\"\\t\"}{if ($5==\"+\") print}'  ncbiRefSeqCurated_NM.promotersUpstream2000_withChromLengths.txt > ncbiRefSeqCurated_NM.promotersUpstream2000_withChromLengths.plusStrand.txt\n",
    "# Use the last col (Chromosome length) to compare ChrEnd value +1000\n",
    "awk 'BEGIN{OFS=FS=\"\\t\"}{if (($3+1000)<$6) print $1,$2,($3+1000),$4,$5; else print $1,$2,$6,$4,$5}' ncbiRefSeqCurated_NM.promotersUpstream2000_withChromLengths.plusStrand.txt > ncbiRefSeqCurated_NM.promotersUpstream2000.plusStrand.downstream1000.txt\n",
    "\n",
    "#\t\t(3c) - strand; Subtract 1000, check if this passes the genome boundary, and if so, print that \n",
    "awk 'BEGIN{OFS=FS=\"\\t\"}{if ($5==\"-\") print}'  ncbiRefSeqCurated_NM.promotersUpstream2000_withChromLengths.txt > ncbiRefSeqCurated_NM.promotersUpstream2000_withChromLengths.minusStrand.txt\n",
    "# Check to make sure adding 1000 downtream of a minus strand promoter won't go below 0\n",
    "awk 'BEGIN{OFS=FS=\"\\t\"}{if (($2-1000)>0) print $1,($2-1000),$3,$4,$5; else print $1,0,$3,$4,$5}' ncbiRefSeqCurated_NM.promotersUpstream2000_withChromLengths.minusStrand.txt > ncbiRefSeqCurated_NM.promotersUpstream2000.minusStrand.downstream1000.txt\n",
    "\n",
    "### (4) Combine [+] and [-] strands\n",
    "cat ncbiRefSeqCurated_NM.promotersUpstream2000.plusStrand.downstream1000.txt ncbiRefSeqCurated_NM.promotersUpstream2000.minusStrand.downstream1000.txt | bedtools sort -i - > ncbiRefSeqCurated_NM.promoters.Upstream2000Downstream1000.txt\n",
    "\n",
    "\n",
    "\n",
    "#### Get exons\n",
    "# Starting file: ncbiRefSeqCurated_exonsBED.txt \n",
    "# (1) Add ID column baed on the current ID column so that IDs match across files (e.g. NM_001308203.2_exon_0_0_chr1_66999276_f --> NM_001308203); truncate after second _\n",
    "# (2) Selet rows(exons) that correspond to \"NM\" genes as per the ID\n",
    "paste ncbiRefSeqCurated_exonsBED.txt <(awk 'BEGIN{OFS=FS=\"\\t\"}{print $4}' ncbiRefSeqCurated_exonsBED.txt | awk 'BEGIN{OFS=\"_\";FS=\"_\"}{print $1,$2}' - ) > ncbiRefSeqCurated_exonsBED.IDcolAdded.txt # 774,7442\n",
    "# Filter out extras cols and rows that are not NM genes\n",
    "awk 'BEGIN{OFS=FS=\"\\t\"}{if ($7~/NM/) print $1,$2,$3,$7}' ncbiRefSeqCurated_exonsBED.IDcolAdded.txt > ncbiRefSeqCurated_NM.exons.txt # 668009\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "#### Combine promoters and exons\n",
    "cat <(awk 'BEGIN{OFS=FS=\"\\t\"}{print $1,$2,$3,$4}' ncbiRefSeqCurated_NM.promoters.Upstream2000Downstream1000.txt) ncbiRefSeqCurated_NM.exons.txt | bedtools sort -i > ncbiRefSeqCurated_NM.promotersUp2000Down1000AndExons.txt \n",
    "\n",
    "\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0aa93759",
   "metadata": {},
   "source": [
    "# Download and format HMRs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "05260dc6",
   "metadata": {},
   "outputs": [],
   "source": [
    "# DOWNLOADS\n",
    "cd /data/hodges_lab/Tim/finalAnalyses_HMRs/hp_lineage/startingBEDs\n",
    "\n",
    "# [ CT ] - [ Coverage % ]\n",
    "# Adrenal - 71.558\n",
    "# http://smithdata.usc.edu/methbase/data/Roadmap-Human-2015/Human_Adrenal-gland/tracks_hg19/\n",
    "# wget http://smithdata.usc.edu/methbase/data/Roadmap-Human-2015/Human_Adrenal-gland/tracks_hg19/Human_Adrenal-gland.hmr.bb\n",
    "\n",
    "# B cell - 11.855\n",
    "# http://smithdata.usc.edu/methbase/data/Hodges-Human-2011/Human_BCell/tracks_hg19/\n",
    "# wget http://smithdata.usc.edu/methbase/data/Hodges-Human-2011/Human_BCell/tracks_hg19/Human_BCell.hmr.bb\n",
    "\n",
    "# Fetal Heart - 37.134\n",
    "# http://smithdata.usc.edu/methbase/data/Roadmap-Human-2015/Human_Fetal-Heart/tracks_hg19/\n",
    "# wget http://smithdata.usc.edu/methbase/data/Roadmap-Human-2015/Human_Fetal-Heart/tracks_hg19/Human_Fetal-Heart.hmr.bb\n",
    "\n",
    "# Fetal Spinal - 33.623\n",
    "# http://smithdata.usc.edu/methbase/data/Roadmap-Human-2015/Human_Fetal-Spinal-Cord/tracks_hg19/\n",
    "# wget http://smithdata.usc.edu/methbase/data/Roadmap-Human-2015/Human_Fetal-Spinal-Cord/tracks_hg19/Human_Fetal-Spinal-Cord.hmr.bb\n",
    "\n",
    "# H1ESC - 25.933\n",
    "# http://smithdata.usc.edu/methbase/data/Lister-ESC-2009/Human_H1ESC/tracks_hg19/\n",
    "# wget http://smithdata.usc.edu/methbase/data/Lister-ESC-2009/Human_H1ESC/tracks_hg19/Human_H1ESC.hmr.bb\n",
    "\n",
    "# Liver - 49.478\n",
    "# http://smithdata.usc.edu/methbase/data/Roadmap-Human-2015/Human_Liver/tracks_hg19/\n",
    "# wget http://smithdata.usc.edu/methbase/data/Roadmap-Human-2015/Human_Liver/tracks_hg19/Human_Liver.hmr.bb\n",
    "\n",
    "# Macrophage - 36.130\n",
    "# http://smithdata.usc.edu/methbase/data/Roadmap-Human-2015/Human_Macrophage/tracks_hg19/\n",
    "# wget http://smithdata.usc.edu/methbase/data/Roadmap-Human-2015/Human_Macrophage/tracks_hg19/Human_Macrophage.hmr.bb\n",
    "\n",
    "# Neutrophil - 11.329\n",
    "# http://smithdata.usc.edu/methbase/data/Hodges-Human-2011/Human_Neut/tracks_hg19/\n",
    "# wget http://smithdata.usc.edu/methbase/data/Hodges-Human-2011/Human_Neut/tracks_hg19/Human_Neut.hmr.bb\n",
    "\n",
    "# T cell - 11.855\n",
    "# http://smithdata.usc.edu/methbase/data/Roadmap-Human-2015/Human_Tcell/tracks_hg19/\n",
    "# wget http://smithdata.usc.edu/methbase/data/Roadmap-Human-2015/Human_Tcell/tracks_hg19/Human_Tcell.hmr.bb\n",
    "\n",
    "# HSC - 37.562\n",
    "# http://smithdata.usc.edu/methbase/data/Roadmap-Human-2015/Human_Tcell/tracks_hg19/\n",
    "# wget http://smithdata.usc.edu/methbase/data/Roadmap-Human-2015/Human_HSC/tracks_hg19/Human_HSC.hmr.bb\n",
    "\n",
    "# Inconsistent 3-5 columns\n",
    "# Convert to Bed\n",
    "./bigBedToBed Human_Adrenal-gland.hmr.bb Human_Adrenal-gland.hmr.bed\n",
    "./bigBedToBed Human_BCell.hmr.bb Human_BCell.hmr.bed\n",
    "./bigBedToBed Human_Fetal-Heart.hmr.bb Human_Fetal-Heart.hmr.bed\n",
    "./bigBedToBed Human_Fetal-Spinal-Cord.hmr.bb Human_Fetal-Spinal-Cord.hmr.bed\n",
    "./bigBedToBed Human_H1ESC.hmr.bb Human_H1ESC.hmr.bed\n",
    "./bigBedToBed Human_Liver.hmr.bb Human_Liver.hmr.bed\n",
    "./bigBedToBed Human_Macrophage.hmr.bb Human_Macrophage.hmr.bed\n",
    "./bigBedToBed Human_Neut.hmr.bb Human_Neut.hmr.bed\n",
    "./bigBedToBed Human_Tcell.hmr.bb Human_Tcell.hmr.bed\n",
    "./bigBedToBed Human_HSC.hmr.bb Human_HSC.hmr.bed\n",
    "\n",
    "\n",
    "# Trim to 3 columns\n",
    "awk 'BEGIN{OFS=FS=\"\\t\"}{print $1,$2,$3}' Human_Adrenal-gland.hmr.bed > Adrenal.hmr.bed\n",
    "awk 'BEGIN{OFS=FS=\"\\t\"}{print $1,$2,$3}' Human_BCell.hmr.bed > Bcell.hmr.bed\n",
    "awk 'BEGIN{OFS=FS=\"\\t\"}{print $1,$2,$3}' Human_Fetal-Heart.hmr.bed > fHeart.hmr.bed\n",
    "awk 'BEGIN{OFS=FS=\"\\t\"}{print $1,$2,$3}' Human_Fetal-Spinal-Cord.hmr.bed > fSpinal.hmr.bed\n",
    "awk 'BEGIN{OFS=FS=\"\\t\"}{print $1,$2,$3}' Human_H1ESC.hmr.bed > H1ESC.hmr.bed\n",
    "awk 'BEGIN{OFS=FS=\"\\t\"}{print $1,$2,$3}' Human_Liver.hmr.bed > Liver.hmr.bed\n",
    "awk 'BEGIN{OFS=FS=\"\\t\"}{print $1,$2,$3}' Human_Macrophage.hmr.bed > Macrophage.hmr.bed\n",
    "awk 'BEGIN{OFS=FS=\"\\t\"}{print $1,$2,$3}' Human_Neut.hmr.bed > Neutrophil.hmr.bed\n",
    "awk 'BEGIN{OFS=FS=\"\\t\"}{print $1,$2,$3}' Human_Tcell.hmr.bed > Tcell.hmr.bed\n",
    "awk 'BEGIN{OFS=FS=\"\\t\"}{print $1,$2,$3}' Human_HSC.hmr.bed > HSC.hmr.bed\n",
    "\n",
    "\n",
    "# Add a size column as a new file\n",
    "awk 'BEGIN{OFS=FS=\"\\t\"}{print $1,$2,$3,($3-$2)}' Adrenal.hmr.bed > Adrenal.hmr.sizes.bed\n",
    "awk 'BEGIN{OFS=FS=\"\\t\"}{print $1,$2,$3,($3-$2)}' Bcell.hmr.bed > Bcell.hmr.sizes.bed\n",
    "awk 'BEGIN{OFS=FS=\"\\t\"}{print $1,$2,$3,($3-$2)}' fHeart.hmr.bed > fHeart.hmr.sizes.bed\n",
    "awk 'BEGIN{OFS=FS=\"\\t\"}{print $1,$2,$3,($3-$2)}' fSpinal.hmr.bed > fSpinal.hmr.sizes.bed\n",
    "awk 'BEGIN{OFS=FS=\"\\t\"}{print $1,$2,$3,($3-$2)}' H1ESC.hmr.bed > H1ESC.hmr.sizes.bed\n",
    "awk 'BEGIN{OFS=FS=\"\\t\"}{print $1,$2,$3,($3-$2)}' Liver.hmr.bed > Liver.hmr.sizes.bed\n",
    "awk 'BEGIN{OFS=FS=\"\\t\"}{print $1,$2,$3,($3-$2)}' Macrophage.hmr.bed > Macrophage.hmr.sizes.bed\n",
    "awk 'BEGIN{OFS=FS=\"\\t\"}{print $1,$2,$3,($3-$2)}' Neutrophil.hmr.bed > Neutrophil.hmr.sizes.bed\n",
    "awk 'BEGIN{OFS=FS=\"\\t\"}{print $1,$2,$3,($3-$2)}' Tcell.hmr.bed > Tcell.hmr.sizes.bed\n",
    "awk 'BEGIN{OFS=FS=\"\\t\"}{print $1,$2,$3,($3-$2)}' HSC.hmr.bed > HSC.hmr.sizes.bed\n",
    "\n",
    "\n",
    "# Look at minimum sizes\n",
    "for CELL in Adrenal Bcell fHeart fSpinal H1ESC Liver Macrophage Neutrophil Tcell\n",
    "do\n",
    "echo ${CELL}\n",
    "sort -k 4n ${CELL}.hmr.sizes.bed | head -n 10\n",
    "done\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "45da3c9f",
   "metadata": {},
   "source": [
    "## Make Hematopoietic lineage files if needed "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "393624dd",
   "metadata": {},
   "outputs": [],
   "source": [
    "# 2_01_make_hplineage_filtsizeandrefseq_files_hpl.sh\n",
    "# short script to combine B/T, M/N, and M/N/T/B\n",
    "\n",
    "# COMBINE HMR.BED FILES\n",
    "# Directory and in_files:\n",
    "HMRBED_DIR=/data/hodges_lab/Tim/finalAnalyses_HMRs/hp_lineage/startingBEDs\n",
    "BCELL_HMRBEDFILE=${HMRBED_DIR}/Bcell.hmr.bed\n",
    "TCELL_HMRBEDFILE=${HMRBED_DIR}/Tcell.hmr.bed\n",
    "MACRO_HMRBEDFILE=${HMRBED_DIR}/Macrophage.hmr.bed\n",
    "NEUT_HMRBEDFILE=${HMRBED_DIR}/Neutrophil.hmr.bed\n",
    "\n",
    "# Out_files:\n",
    "BT_HMRBEDOUTFILE=${HMRBED_DIR}/BT.hmr.bed\n",
    "MN_HMRBEDOUTFILE=${HMRBED_DIR}/MN.hmr.bed\n",
    "MNBT_HMRBEDOUTFILE=${HMRBED_DIR}/MNBT.hmr.bed\n",
    "\n",
    "\n",
    "# B/T\n",
    "cat ${BCELL_HMRBEDFILE} ${TCELL_HMRBEDFILE} | bedtools sort -i - | bedtools merge -c 2 -o count -i - | awk 'BEGIN{OFS=FS=\"\\t\"}{if ($4>1) print}' - > ${BT_HMRBEDOUTFILE}\n",
    "\n",
    "# M/N\n",
    "cat ${MACRO_HMRBEDFILE} ${NEUT_HMRBEDFILE} | bedtools sort -i - | bedtools merge -c 2 -o count -i - | awk 'BEGIN{OFS=FS=\"\\t\"}{if ($4>1) print}' - > ${MN_HMRBEDOUTFILE}\n",
    "\n",
    "# M/N/B/T\n",
    "awk 'BEGIN{OFS=FS=\"\\t\"}{print $0,\"Bcell\"}' ${BCELL_HMRBEDFILE} > ${BCELL_HMRBEDFILE}.temp\n",
    "awk 'BEGIN{OFS=FS=\"\\t\"}{print $0,\"Tcell\"}' ${TCELL_HMRBEDFILE} > ${TCELL_HMRBEDFILE}.temp\n",
    "awk 'BEGIN{OFS=FS=\"\\t\"}{print $0,\"Macro\"}' ${MACRO_HMRBEDFILE} > ${MACRO_HMRBEDFILE}.temp\n",
    "awk 'BEGIN{OFS=FS=\"\\t\"}{print $0,\"Neut\"}' ${NEUT_HMRBEDFILE} > ${NEUT_HMRBEDFILE}.temp\n",
    "cat ${BCELL_HMRBEDFILE}.temp ${TCELL_HMRBEDFILE}.temp ${MACRO_HMRBEDFILE}.temp ${NEUT_HMRBEDFILE}.temp | bedtools sort -i - | bedtools merge -c 4 -o count_distinct -i - | awk 'BEGIN{OFS=FS=\"\\t\"}{if ($4>3) print}' - > ${MNBT_HMRBEDOUTFILE}\n",
    "\n",
    "\n",
    "\n",
    "awk 'BEGIN{OFS=FS=\"\\t\"}{print $0,Bcell}' ${BCELL_HMRBEDFILE} | head\n",
    "\n",
    "\n",
    "\n",
    "# COMBINE FILT REF SEQ FILES\n",
    "# Directory and in_files:\n",
    "HMR_DIR=/data/hodges_lab/Tim/finalAnalyses_HMRs/hp_lineage/filtRefSeqFiles\n",
    "BCELL_FILE=${HMR_DIR}/Bcell.minsize50.filtforrefseqTSSexons.txt\n",
    "TCELL_FILE=${HMR_DIR}/Tcell.minsize50.filtforrefseqTSSexons.txt\n",
    "MACRO_FILE=${HMR_DIR}/Macrophage.minsize50.filtforrefseqTSSexons.txt\n",
    "NEUT_FILE=${HMR_DIR}/Neutrophil.minsize50.filtforrefseqTSSexons.txt\n",
    "\n",
    "# Out_files:\n",
    "BT_OUTFILE=${HMR_DIR}/BT.minsize50.filtforrefseqTSSexons.txt\n",
    "MN_OUTFILE=${HMR_DIR}/MN.minsize50.filtforrefseqTSSexons.txt\n",
    "MNBT_OUTFILE=${HMR_DIR}/MNBT.minsize50.filtforrefseqTSSexons.txt\n",
    "\n",
    "\n",
    "# B/T\n",
    "cat ${BCELL_FILE} ${TCELL_FILE} | bedtools sort -i - | bedtools merge -c 2 -o count -i - | awk 'BEGIN{OFS=FS=\"\\t\"}{if ($4>1) print}' - > ${BT_OUTFILE}\n",
    "\n",
    "# M/N\n",
    "cat ${MACRO_FILE} ${NEUT_FILE} | bedtools sort -i - | bedtools merge -c 2 -o count -i - | awk 'BEGIN{OFS=FS=\"\\t\"}{if ($4>1) print}' - > ${MN_OUTFILE}\n",
    "\n",
    "# M/N/B/T\n",
    "awk 'BEGIN{OFS=FS=\"\\t\"}{print $0,\"Bcell\"}' ${BCELL_FILE} > ${BCELL_FILE}.temp\n",
    "awk 'BEGIN{OFS=FS=\"\\t\"}{print $0,\"Tcell\"}' ${TCELL_FILE} > ${TCELL_FILE}.temp\n",
    "awk 'BEGIN{OFS=FS=\"\\t\"}{print $0,\"Macro\"}' ${MACRO_FILE} > ${MACRO_FILE}.temp\n",
    "awk 'BEGIN{OFS=FS=\"\\t\"}{print $0,\"Neut\"}' ${NEUT_FILE} > ${NEUT_FILE}.temp\n",
    "cat ${BCELL_FILE}.temp ${TCELL_FILE}.temp ${MACRO_FILE}.temp ${NEUT_FILE}.temp | bedtools sort -i - | bedtools merge -c 4 -o count_distinct -i - | awk 'BEGIN{OFS=FS=\"\\t\"}{if ($4>3) print}' - > ${MNBT_OUTFILE}\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6716d9e3",
   "metadata": {},
   "source": [
    "# Filter size and RefSeq "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a4a92000",
   "metadata": {},
   "source": [
    "## Script code"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7f2661a9",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Script to size filter, then cross ref vs RefSeq TSS/Exons\n",
    "# 2 steps: #1 filt size, #2 filt for refseq\n",
    "# usage: sh filtsizeandrefseq.sh [file] [size]; e.g. sh filtsizeandrefseq.sh Bcell.hmr.bed 50\n",
    "\n",
    "# Variables:\n",
    "infile=\"$1\"\n",
    "size=\"$2\"\n",
    "infileDir=\"$3\"\n",
    "outfileDir=\"$4\"\n",
    "\n",
    "# Directories:\n",
    "referenceDir=/data/hodges_lab/Tim/finalAnalyses_HMRs/hp_lineage/referenceFiles\n",
    "\n",
    "# File names used within this script\n",
    "infileName=${infile%%.*}\n",
    "outName=$infileName.minsize$size.filtforrefseqTSSexons.txt\n",
    "\n",
    "\n",
    "# Filter by minimum size | Filter by promoters and exons\n",
    "awk -v size=\"$size\" 'BEGIN{OFS=FS=\"\\t\"}{if (($3-$2)>(size-1)) {print $1,$2,$3,($3-$2)}}' $infileDir/$infile | bedtools intersect -v -a - -b $referenceDir/ncbiRefSeqCurated_NM.promotersAndExons.txt > $outfileDir/$outName\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8bf4ddb9",
   "metadata": {},
   "source": [
    "## Running script loop"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1562f55f",
   "metadata": {},
   "outputs": [],
   "source": [
    "# 2_run_filtsizeandrefseq_acrossCelltypes_hpl.sh\n",
    "# Simple script to automate passing all CTs through the filtsizeandrefseq script\n",
    "\n",
    "# Uses: \n",
    "# \t- filtsizeandrefseq.sh\n",
    "\n",
    "# Variables:\n",
    "SCRIPT_DIR=/data/hodges_lab/Tim/finalAnalyses_HMRs/hp_lineage/scripts/helperScripts\n",
    "IN_DIR=/data/hodges_lab/Tim/finalAnalyses_HMRs/hp_lineage/startingBEDs\n",
    "OUT_DIR=/data/hodges_lab/Tim/finalAnalyses_HMRs/hp_lineage/filtRefSeqFiles # Currently handled by the filtsizeandrefseq script.\n",
    "\n",
    "for CELLTYPE in Adrenal Bcell fHeart fSpinal H1ESC Liver Macrophage Neutrophil Tcell HSC\n",
    "do\n",
    "\tsh ${SCRIPT_DIR}/filtsizeandrefseq_hpl.sh ${CELLTYPE}.hmr.bed 50 ${IN_DIR} ${OUT_DIR}\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "482e42b2",
   "metadata": {},
   "source": [
    "# Make anti files"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2f830132",
   "metadata": {},
   "source": [
    "## Script code"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9d66c6fa",
   "metadata": {},
   "outputs": [],
   "source": [
    "# make_anti.sh\n",
    "# Usage: sh make_anti.sh [anchor CT name] [other cell type HMR files]\n",
    "# Usage example: sh make_anti.sh Adrenal Bcell.hmr.bed fHeart.hmr.bed [...]\n",
    "# This script is designed to take any length of \"other\" comparison inputs\n",
    "\n",
    "infilesDir=$1\n",
    "outfileDir=$2\n",
    "Anchor_CT=$3\n",
    "shift 3\n",
    "Other_Files=\"$@\"\n",
    "\n",
    "# Cat all the other files together | sort this list | merge to simplify regions with high overlap \n",
    "cd $infilesDir\n",
    "cat $Other_Files | bedtools sort -i - | bedtools merge -i - > $outfileDir/anti_${Anchor_CT}\\_hpl.txt\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "976c15b1",
   "metadata": {},
   "source": [
    "## Code to run script on all cell types"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "01b94015",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Uses: \n",
    "# \t- filtsizeandrefseq.sh\n",
    "\n",
    "# Usage: sh make_anti.sh [anchor CT name] [other cell type HMR files]\n",
    "# Usage example: sh make_anti.sh Adrenal Bcell.hmr.bed fHeart.hmr.bed [...]\n",
    "SCRIPT_DIR=/data/hodges_lab/Tim/finalAnalyses_HMRs/hp_lineage/scripts/helperScripts\n",
    "IN_DIR=/data/hodges_lab/Tim/finalAnalyses_HMRs/hp_lineage/filtRefSeqFiles\n",
    "OUT_DIR=/data/hodges_lab/Tim/finalAnalyses_HMRs/hp_lineage/antiFiles\n",
    "\n",
    "sh ${SCRIPT_DIR}/make_anti_hpl.sh ${IN_DIR} ${OUT_DIR} Adrenal Bcell.minsize50.filtforrefseqTSSexons.txt fSpinal.minsize50.filtforrefseqTSSexons.txt H1ESC.minsize50.filtforrefseqTSSexons.txt Liver.minsize50.filtforrefseqTSSexons.txt fHeart.minsize50.filtforrefseqTSSexons.txt Macrophage.minsize50.filtforrefseqTSSexons.txt Neutrophil.minsize50.filtforrefseqTSSexons.txt Tcell.minsize50.filtforrefseqTSSexons.txt HSC.minsize50.filtforrefseqTSSexons.txt\n",
    "\n",
    "sh ${SCRIPT_DIR}/make_anti_hpl.sh ${IN_DIR} ${OUT_DIR} Bcell Adrenal.minsize50.filtforrefseqTSSexons.txt fSpinal.minsize50.filtforrefseqTSSexons.txt H1ESC.minsize50.filtforrefseqTSSexons.txt Liver.minsize50.filtforrefseqTSSexons.txt fHeart.minsize50.filtforrefseqTSSexons.txt Macrophage.minsize50.filtforrefseqTSSexons.txt Neutrophil.minsize50.filtforrefseqTSSexons.txt Tcell.minsize50.filtforrefseqTSSexons.txt HSC.minsize50.filtforrefseqTSSexons.txt\n",
    "\n",
    "sh ${SCRIPT_DIR}/make_anti_hpl.sh ${IN_DIR} ${OUT_DIR} fHeart Adrenal.minsize50.filtforrefseqTSSexons.txt Bcell.minsize50.filtforrefseqTSSexons.txt fSpinal.minsize50.filtforrefseqTSSexons.txt H1ESC.minsize50.filtforrefseqTSSexons.txt Liver.minsize50.filtforrefseqTSSexons.txt Macrophage.minsize50.filtforrefseqTSSexons.txt Neutrophil.minsize50.filtforrefseqTSSexons.txt Tcell.minsize50.filtforrefseqTSSexons.txt HSC.minsize50.filtforrefseqTSSexons.txt\n",
    "\n",
    "sh ${SCRIPT_DIR}/make_anti_hpl.sh ${IN_DIR} ${OUT_DIR} fSpinal Adrenal.minsize50.filtforrefseqTSSexons.txt Bcell.minsize50.filtforrefseqTSSexons.txt H1ESC.minsize50.filtforrefseqTSSexons.txt Liver.minsize50.filtforrefseqTSSexons.txt fHeart.minsize50.filtforrefseqTSSexons.txt Macrophage.minsize50.filtforrefseqTSSexons.txt Neutrophil.minsize50.filtforrefseqTSSexons.txt Tcell.minsize50.filtforrefseqTSSexons.txt HSC.minsize50.filtforrefseqTSSexons.txt\n",
    "\n",
    "sh ${SCRIPT_DIR}/make_anti_hpl.sh ${IN_DIR} ${OUT_DIR} H1ESC Adrenal.minsize50.filtforrefseqTSSexons.txt Bcell.minsize50.filtforrefseqTSSexons.txt fSpinal.minsize50.filtforrefseqTSSexons.txt Liver.minsize50.filtforrefseqTSSexons.txt fHeart.minsize50.filtforrefseqTSSexons.txt Macrophage.minsize50.filtforrefseqTSSexons.txt Neutrophil.minsize50.filtforrefseqTSSexons.txt Tcell.minsize50.filtforrefseqTSSexons.txt HSC.minsize50.filtforrefseqTSSexons.txt\n",
    "\n",
    "sh ${SCRIPT_DIR}/make_anti_hpl.sh ${IN_DIR} ${OUT_DIR} Liver Adrenal.minsize50.filtforrefseqTSSexons.txt Bcell.minsize50.filtforrefseqTSSexons.txt fSpinal.minsize50.filtforrefseqTSSexons.txt H1ESC.minsize50.filtforrefseqTSSexons.txt fHeart.minsize50.filtforrefseqTSSexons.txt Macrophage.minsize50.filtforrefseqTSSexons.txt Neutrophil.minsize50.filtforrefseqTSSexons.txt Tcell.minsize50.filtforrefseqTSSexons.txt HSC.minsize50.filtforrefseqTSSexons.txt\n",
    "\n",
    "sh ${SCRIPT_DIR}/make_anti_hpl.sh ${IN_DIR} ${OUT_DIR} Macrophage Adrenal.minsize50.filtforrefseqTSSexons.txt Bcell.minsize50.filtforrefseqTSSexons.txt fSpinal.minsize50.filtforrefseqTSSexons.txt H1ESC.minsize50.filtforrefseqTSSexons.txt fHeart.minsize50.filtforrefseqTSSexons.txt Liver.minsize50.filtforrefseqTSSexons.txt Neutrophil.minsize50.filtforrefseqTSSexons.txt Tcell.minsize50.filtforrefseqTSSexons.txt HSC.minsize50.filtforrefseqTSSexons.txt\n",
    "\n",
    "sh ${SCRIPT_DIR}/make_anti_hpl.sh ${IN_DIR} ${OUT_DIR} Neutrophil Adrenal.minsize50.filtforrefseqTSSexons.txt Bcell.minsize50.filtforrefseqTSSexons.txt fSpinal.minsize50.filtforrefseqTSSexons.txt H1ESC.minsize50.filtforrefseqTSSexons.txt fHeart.minsize50.filtforrefseqTSSexons.txt Liver.minsize50.filtforrefseqTSSexons.txt Macrophage.minsize50.filtforrefseqTSSexons.txt Tcell.minsize50.filtforrefseqTSSexons.txt HSC.minsize50.filtforrefseqTSSexons.txt\n",
    "\n",
    "sh ${SCRIPT_DIR}/make_anti_hpl.sh ${IN_DIR} ${OUT_DIR} Tcell Adrenal.minsize50.filtforrefseqTSSexons.txt Bcell.minsize50.filtforrefseqTSSexons.txt fSpinal.minsize50.filtforrefseqTSSexons.txt H1ESC.minsize50.filtforrefseqTSSexons.txt fHeart.minsize50.filtforrefseqTSSexons.txt Liver.minsize50.filtforrefseqTSSexons.txt Macrophage.minsize50.filtforrefseqTSSexons.txt Neutrophil.minsize50.filtforrefseqTSSexons.txt HSC.minsize50.filtforrefseqTSSexons.txt\n",
    "\n",
    "sh ${SCRIPT_DIR}/make_anti_hpl.sh ${IN_DIR} ${OUT_DIR} HSC Adrenal.minsize50.filtforrefseqTSSexons.txt Bcell.minsize50.filtforrefseqTSSexons.txt fSpinal.minsize50.filtforrefseqTSSexons.txt H1ESC.minsize50.filtforrefseqTSSexons.txt fHeart.minsize50.filtforrefseqTSSexons.txt Liver.minsize50.filtforrefseqTSSexons.txt Macrophage.minsize50.filtforrefseqTSSexons.txt Neutrophil.minsize50.filtforrefseqTSSexons.txt Tcell.minsize50.filtforrefseqTSSexons.txt\n",
    "\n",
    "# HP LINEAGE COMBINATIONS\n",
    "# B/T\n",
    "sh ${SCRIPT_DIR}/make_anti_hpl.sh ${IN_DIR} ${OUT_DIR} BT Adrenal.minsize50.filtforrefseqTSSexons.txt fSpinal.minsize50.filtforrefseqTSSexons.txt H1ESC.minsize50.filtforrefseqTSSexons.txt Liver.minsize50.filtforrefseqTSSexons.txt fHeart.minsize50.filtforrefseqTSSexons.txt Macrophage.minsize50.filtforrefseqTSSexons.txt Neutrophil.minsize50.filtforrefseqTSSexons.txt HSC.minsize50.filtforrefseqTSSexons.txt\n",
    "\n",
    "# M/N\n",
    "sh ${SCRIPT_DIR}/make_anti_hpl.sh ${IN_DIR} ${OUT_DIR} MN Adrenal.minsize50.filtforrefseqTSSexons.txt Bcell.minsize50.filtforrefseqTSSexons.txt fSpinal.minsize50.filtforrefseqTSSexons.txt H1ESC.minsize50.filtforrefseqTSSexons.txt fHeart.minsize50.filtforrefseqTSSexons.txt Liver.minsize50.filtforrefseqTSSexons.txt  Tcell.minsize50.filtforrefseqTSSexons.txt HSC.minsize50.filtforrefseqTSSexons.txt\n",
    "\n",
    "# M/N/B/T\n",
    "sh ${SCRIPT_DIR}/make_anti_hpl.sh ${IN_DIR} ${OUT_DIR} MNBT Adrenal.minsize50.filtforrefseqTSSexons.txt fSpinal.minsize50.filtforrefseqTSSexons.txt H1ESC.minsize50.filtforrefseqTSSexons.txt Liver.minsize50.filtforrefseqTSSexons.txt fHeart.minsize50.filtforrefseqTSSexons.txt HSC.minsize50.filtforrefseqTSSexons.txt\n",
    "\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3400ef41",
   "metadata": {},
   "source": [
    "# Run annotation script"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6f60fa3a",
   "metadata": {},
   "source": [
    "## Script code"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4b9a821a",
   "metadata": {},
   "outputs": [],
   "source": [
    "# This script is designed to \n",
    "# Usage: sh annotate_HMRs.sh [CT] [CT BED file] [anti-CT BED file] [Blacklist regions for custom merge - \"internal clusters\"]  \n",
    "# Example usage: sh annotate_HMRs_totalClusters.sh Adrenal Adrenal.minsize50.filtforrefseqTSSexons.txt anti_Adrenal.txt /data/hodges_lab/Tim/finalAnalyses_HMRs/HMRs_goodGenes/referenceFiles/ncbiRefSeqCurated_NM.promotersAndExons.txt /data/hodges_lab/Tim/finalAnalyses_HMRs/HMRs_goodGenes/HMRs /data/hodges_lab/Tim/finalAnalyses_HMRs/HMRs_goodGenes/filtRefSeqFiles /data/hodges_lab/Tim/finalAnalyses_HMRs/HMRs_goodGenes/antiFiles\n",
    "\n",
    "\n",
    "# INPUT: \n",
    "# (1) CT Name # for naming conventions\n",
    "# (2) CT BED file\n",
    "# (3) Anti-CT file\n",
    "# (4) Reference file for blacklist\n",
    "# (5) Output directory\n",
    "# (6) Input directory # generally filtRefSeq dir\n",
    "# (7) Anti-CT directory\n",
    "\n",
    "\n",
    "# OUTPUT (11): Clustered/Unclustered files further delineated by cell-type specificity - up to four different levels of celltype membership binning\n",
    "# (0) All_cellspecific\n",
    "# (00) All_shared\n",
    "\n",
    "# (1) CT_unclustered\n",
    "# (2) CT_unclustered_cellspecific\n",
    "# (3) CT_unclustered_shared\n",
    "\n",
    "# (4) CT_clustered_endToEnd\n",
    "# (5) CT_clustered_individualHMRs\n",
    "# (6) CT_clustered_individualHMRs_cellspecific\n",
    "# (7) CT_clustered_individualHMRs_shared\n",
    "# (8) internalClusters_containsCS\n",
    "# (9) internalClusters_withoutCS\n",
    "\n",
    "\n",
    "\n",
    "######################       ISSUE REGIONS       ############################\n",
    "# chr10\t22623401\t22623588 - Says Lineage, but one of those with HMRs spannign TSS/Exon\n",
    "# Remake \"Other\" file:\n",
    "#\t\t\tcat Rvent.HMR.min50.txt Liver.HMR.min50.txt H1ESC.HMR.min50.txt | bedtools sort -i > Rvent.Liver.H1ESC.min50.txt\n",
    "\n",
    "################################################################################\n",
    "\n",
    "# Variables #Expects 5 cell type BED files as input\n",
    "CT=$1\n",
    "CTFile=$2 # From inDir\n",
    "OthFile=$3 # e.g. Anti_Bcell.bed; a collection of regions from all other CTs\n",
    "TSSExonRef=$4 # i.e. blacklist file for merging boundaries in establishing clusters\n",
    "outDir=$5\n",
    "inDir=$6\n",
    "antiDir=$7\n",
    "\n",
    "\n",
    "# Echo the variable assignments for good measure\n",
    "echo \"Input Directory identified: ${inDir}\"\n",
    "echo \"Output Directory identified: ${outDir}\"\n",
    "echo \"Cell-type identified: ${CT}\"\n",
    "echo \"Cell-type file identified: ${CTFile}\"\n",
    "echo \"Other cell-type file identified: ${OthFile}\"\n",
    "echo \"Blacklist found: ${TSSExonRef}\"\n",
    "echo \"\"\n",
    "echo \"\"\n",
    "echo \"\"\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "################################################################################\n",
    "# Find ALL HMRs\n",
    "echo \"Finding all cell specific and shared HMRs\"\n",
    "\n",
    "##### ((0)) ALL_cellspecific\n",
    "##### Exclude HMRs overlapping with blacklist | Exclude HMRs from this celltype that overlap with those from other celltypes\n",
    "bedtools intersect -v -a $inDir/$CTFile -b $TSSExonRef | bedtools intersect -v -a - -b $antiDir/$OthFile > $outDir/$CT\\_all_cellspecific_hpl.txt\n",
    "echo \"Found cell-specific HMRs.\"\n",
    "\n",
    "##### ((00)) ALL_shared\n",
    "##### Exclude HMRs overlapping with blacklist | Include HMRs from this celltype that overlap with those from other celltypes\n",
    "bedtools intersect -v -a $inDir/$CTFile -b $TSSExonRef | bedtools intersect -u -a - -b $antiDir/$OthFile > $outDir/$CT\\_all_shared_hpl.txt\n",
    "echo \"Found shared HMRs.\"\n",
    "echo \"\"\n",
    "echo \"\"\n",
    "echo \"\"\n",
    "\n",
    "\n",
    "\n",
    "################################################################################\n",
    "# Find Unclustered HMRs\n",
    "\n",
    "##### ((1)) CT_unclustered\n",
    "#####  Filt for TSS/Exons | Merge together clustered HMRs based on rules | Filter out unclustered\n",
    "originalHMRDir=/data/hodges_lab/Tim/finalAnalyses_HMRs/hp_lineage/startingBEDs/\n",
    "startingFile=${originalHMRDir}${CT}.hmr.bed\n",
    "\n",
    "awk 'BEGIN{OFS=FS=\"\\t\"}{if (($3-$2)>49) print $1,$2,$3}' ${startingFile} | bedtools merge -c 2 -o count -d 6000 -i - | awk 'BEGIN{FS=OFS=\"\\t\"}{if ($4<2) print}' - | awk 'BEGIN{OFS=FS=\"\\t\"}{print $1,$2,$3}' - | bedtools intersect -v -a - -b ${TSSExonRef} > $outDir/$CT\\_unclustered_hpl.txt\n",
    "echo \"Found unclustered HMRs.\"\n",
    "\n",
    "\n",
    "##### ((2)) CT_unclustered_cellspecific\n",
    "##### \n",
    "# Inverse intersect (1) with comparison file to find (a) ct-specific unclustered HMRs\n",
    "bedtools intersect -v -a $outDir/$CT\\_unclustered_hpl.txt -b $antiDir/$OthFile | awk 'BEGIN{OFS=FS=\"\\t\"}{print $1,$2,$3}' - > $outDir/$CT\\_unclustered_cellspecific_hpl.txt\n",
    "echo \"Found cell-specific unclustered HMRs.\"\n",
    "\n",
    "\n",
    "##### ((3) CT_unclustered_shared\n",
    "##### \n",
    "# Intersect (1) with comparison file to find (b) ct-shared unclustered HMRs\n",
    "bedtools intersect -u -a $outDir/$CT\\_unclustered_hpl.txt -b $antiDir/$OthFile | awk 'BEGIN{OFS=FS=\"\\t\"}{print $1,$2,$3}' - > $outDir/$CT\\_unclustered_shared_hpl.txt\n",
    "echo \"Found shared unclustered HMRs.\"\n",
    "echo \"\"\n",
    "echo \"\"\n",
    "echo \"\"\n",
    "\n",
    "################################################################################\n",
    "# Find Clustered HMRs\n",
    "\n",
    "##### ((4)) CT_internalClusters\n",
    "##### \n",
    "# internal Clusters that don't cross at TSS/Exon, that also compose 3+ HMRs\n",
    "# Strategy: \n",
    "#\t\t- (pre-a) Create a whitelist from the TSS/Exon blacklist\n",
    "# \t\t- (a) Find all inverse regions (Whitelist regions) that contain 3+ HMRs (potential clusters)\n",
    "# \t\t- (b) Make BED that has two BED coordinates per line - Left: Whitelist Region; Right: HMR region\n",
    "#\t\t- (c) Merge using this altered BED-BED file and a custom awk script instead of BEDtools Merge\n",
    "\n",
    "# (pre-a) Turn blacklist into whitelist\n",
    "# To download the hg19 chr size file: \n",
    "# mysql --user=genome --host=genome-mysql.cse.ucsc.edu -A -e \"select chrom, size from hg19.chromInfo\"  > hg19.genome\n",
    "# awk 'BEGIN{OFS=FS=\"\\t\"}{print $1,$2,$2}' /scratch/scottt7/paper_wases/HMRs/hg19.genome | bedtools sort -i - > /scratch/scottt7/paper_wases/HMRs/hg19.genome_sorted.txt\n",
    "awk 'BEGIN{OFS=FS=\"\\t\"}{if ($1 !~ /\\_/)print $1,$2,$3}' $TSSExonRef | bedtools sort -i - | bedtools complement -i - -g /scratch/scottt7/paper_wases/HMRs/hg19.chrom.sizes.twoCol.sorted > $outDir/$CT\\_whitelist.txt\n",
    "echo \"Made whitelist.\"\n",
    "\n",
    "\n",
    "# (a) Find whitelist regions with 3+ HMRs\n",
    "bedtools intersect -c -F 1.0 -a $outDir/$CT\\_whitelist.txt -b $inDir/$CTFile | awk 'BEGIN{OFS=FS=\"\\t\"}{if ($4>2) print}' - > $outDir/$CT\\_whitelistRegions_contain3ormoreHMRs_hpl.txt\n",
    "echo \"Found whitelist regions that contain 3+ HMRs.\"\n",
    "\n",
    "# (b)Find what HMRs are in these to create a double BED file in a parsable list for custom merging: [1-3]: BED coordinates of Whitelist region [4]: #HMR in Whitelist region, [5-7]: BED coordinates of HMRs in Whitelist region\n",
    "# Diagnostic: bedtools intersect -loj -F 1.0 -a Bcell_whitelistRegions_contain3ormoreHMRs.txt -b Bcell.HMR.min50.txt | awk 'BEGIN{OFS=FS=\"\\t\"}{print}' - | bedtools sort -i - | head\n",
    "# awk 'BEGIN{OFS=FS=\"\\t\"}{print $5,$6,$7,$1,$2,$3,$4}' temp.txt | head\n",
    "\n",
    "bedtools intersect -loj -F 1.0 -a $outDir/$CT\\_whitelistRegions_contain3ormoreHMRs_hpl.txt -b $inDir/$CTFile | awk 'BEGIN{OFS=FS=\"\\t\"}{print $5,$6,$7,$1,$2,$3,$4}' - | bedtools sort -i - > $outDir/$CT\\_whitelistRegions_cluster_individualHMRs_temp_hpl.txt\n",
    "echo \"Created double BED for whitelist regions that contain 3+ HMRs.\"\n",
    "\n",
    "# (c) Custom merge within these regions\n",
    "awk -v whitelistStart=1 -v hmrsCount=1 -v clusterChr=1 -v clusterStart=1 -v clusterEnd=2 'BEGIN{OFS=FS=\"\\t\";dist=6000;whitelistStart=1;hmrsCount=1;clusterChr=1;clusterStart=1;clusterEnd=2;} {\n",
    "\t# Check if we are in the same WhitelistBoundary\n",
    "\tif ($5!=whitelistStart) {\n",
    "\t\tprint clusterChr,clusterStart,clusterEnd,hmrsCount;\n",
    "\t\twhitelistStart=$5;\n",
    "\t\thmrsCount=1;\n",
    "\t\tclusterChr=$1;\n",
    "\t\tclusterStart=$2;\n",
    "\t\tclusterEnd=$3;\n",
    "\t} else {\n",
    "\t\t# case if (1) we are staying within a Whitelist boundary, but (2) the previous HMR was >6000bp away\n",
    "\t\tif (($2-clusterEnd)>6000) {\n",
    "\t\t\tprint clusterChr,clusterStart,clusterEnd,hmrsCount;\n",
    "\t\t\tclusterStart=$2;\n",
    "\t\t\tclusterEnd=$3;\n",
    "\t\t\thmrsCount=1;\n",
    "\t\t} else {\n",
    "\t\t\tclusterEnd=$3;\n",
    "\t\t\thmrsCount+=1;\n",
    "\t\t}\n",
    "\t}\n",
    "}' $outDir/$CT\\_whitelistRegions_cluster_individualHMRs_temp_hpl.txt | awk 'BEGIN{OFS=FS=\"\\t\"}{if ($4>2) print $0}' - > $outDir/$CT\\_internalClusters_hpl.txt\n",
    "echo \"Found internalClusters containing 3+ HMRs.\"\n",
    "\n",
    "#########################\n",
    "#########################\n",
    "#########################\n",
    "# Remove unplaced contigs, causes issues downline \n",
    "awk 'BEGIN{OFS=FS=\"\\t\"}{if ($1 !~ /\\_/)print $1,$2,$3}' $outDir/$CT\\_internalClusters_hpl.txt > $outDir/$CT\\_internalClusters_hpl_BED.txt\n",
    "echo $\"\\n\"\n",
    "\n",
    "\n",
    "##### ((5)) internalClusters_individualHMRs \n",
    "#####\n",
    "bedtools intersect -u -a $inDir/$CTFile -b $outDir/$CT\\_internalClusters_hpl_BED.txt > $outDir/$CT\\_internalClusters_individualHMRs_hpl.txt\n",
    "echo \"Found the HMRs composing internalClusters containing 3+ HMRs.\"\n",
    "\n",
    "##### ((6)) internalClusters_individualHMRs_cellspecific\n",
    "##### \n",
    "# Inverse intersect (1) with our comparison file to find (a) ct-specific and (b) at least partially shared regions\n",
    "bedtools intersect -v -a $outDir/$CT\\_internalClusters_individualHMRs_hpl.txt -b $antiDir/$OthFile > $outDir/$CT\\_internalClusters_individualHMRs_cellspecific_hpl.txt\n",
    "echo \"Found cell-specific clustered HMRs.\"\n",
    "\n",
    "\n",
    "##### ((7) CT_internalClusters_individualHMRs_shared\n",
    "##### \n",
    "#\tIntersect CT with OtherFile to find shared\n",
    "bedtools intersect -u -a $outDir/$CT\\_internalClusters_individualHMRs_hpl.txt -b $antiDir/$OthFile > $outDir/$CT\\_internalClusters_individualHMRs_shared_hpl.txt\n",
    "echo \"Found shared clustered HMRs.\"\n",
    "\n",
    "\n",
    "##### ((8)) internalClusters_containsCS\n",
    "#####\n",
    "bedtools intersect -u -a $outDir/$CT\\_internalClusters_hpl_BED.txt -b $outDir/$CT\\_all_cellspecific_hpl.txt > $outDir/$CT\\_internalClusters_containsCS_hpl.txt\n",
    "echo \"Found the HMRs composing internalClusters containing 3+ HMRs.\"\n",
    "\n",
    "\n",
    "##### ((9)) internalClusters_withoutCS \n",
    "#####\n",
    "bedtools intersect -v -a $outDir/$CT\\_internalClusters_hpl_BED.txt -b $outDir/$CT\\_all_cellspecific_hpl.txt > $outDir/$CT\\_internalClusters_withoutCS_hpl.txt\n",
    "echo \"Found the HMRs composing internalClusters containing 3+ HMRs.\"\n",
    "echo \"\"\n",
    "echo \"\"\n",
    "echo \"\"\n",
    "echo \"\"\n",
    "echo \"\"\n",
    "echo \"\"\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2b8e8120",
   "metadata": {},
   "source": [
    "## Code to run script on cell types"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "86d8f3c9",
   "metadata": {},
   "outputs": [],
   "source": [
    "# annotation_all_code.sh\n",
    "# loop annotate_HMRs over the celltypes\n",
    "\n",
    "SCRIPT_DIR=/data/hodges_lab/Tim/finalAnalyses_HMRs/hp_lineage/scripts/helperScripts\n",
    "REF_DIR=/data/hodges_lab/Tim/finalAnalyses_HMRs/hp_lineage/referenceFiles\n",
    "OUT_DIR=/data/hodges_lab/Tim/finalAnalyses_HMRs/hp_lineage/HMRs\n",
    "IN_DIR=/data/hodges_lab/Tim/finalAnalyses_HMRs/hp_lineage/filtRefSeqFiles\n",
    "ANTI_DIR=/data/hodges_lab/Tim/finalAnalyses_HMRs/hp_lineage/antiFiles\n",
    "\n",
    "\n",
    "# annotate_HMRs.sh\n",
    "# INPUT: \n",
    "# (1) CT Name # for naming conventions\n",
    "# (2) CT BED file (looks for this in input directory)\n",
    "# (3) Anti-CT file\n",
    "# (4) Reference file for blacklist\n",
    "# (5) Output directory\n",
    "# (6) Input directory # generally filtRefSeq dir\n",
    "# (7) Anti-CT directory\n",
    "\n",
    "for CELLTYPE in Adrenal Bcell fHeart fSpinal H1ESC Liver Macrophage Neutrophil Tcell\n",
    "do\n",
    "\tsh ${SCRIPT_DIR}/annotate_HMRs.sh ${CELLTYPE} ${CELLTYPE}.minsize50.filtforrefseqTSSexons.txt \\\n",
    "\tanti_${CELLTYPE}\\_hpl.txt \\\n",
    "\t${REF_DIR}/ncbiRefSeqCurated_NM.promotersUp2000Down1000AndExons.txt \\\n",
    "\t${OUT_DIR} \\\n",
    "\t${IN_DIR} \\\n",
    "\t${ANTI_DIR}\n",
    "done\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4fec59a8",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {},
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": false,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {},
   "toc_section_display": true,
   "toc_window_display": true
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
