{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "8b2286cf",
   "metadata": {},
   "source": [
    "### Want to produce clustered values for various linking distances\n",
    "\n",
    "Distances to try: \n",
    "- 6 kb (Currently have)\n",
    "- 10.5 kb (Reflects approximate mean inter-HMR distance among all HMRs across cell types - \"average\")\n",
    "- 12.5 kb (Super-enhancer/ROSE-popularized definition for linking distance - derived from ChIP-seq data)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "43f65370",
   "metadata": {},
   "source": [
    "### Can modify the custom clustering algorithm down below"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2da9179e",
   "metadata": {},
   "outputs": [],
   "source": [
    "for celltype in Adrenal Bcell fHeart fSpinal H1ESC Liver\n",
    "do\n",
    "\tsh annotate_HMRs.sh ${celltype} ${celltype}.minsize50.filtforrefseqTSSexons.txt anti_${celltype}.txt /data/hodges_lab/Tim/finalAnalyses_HMRs/HMRs_goodGenes/referenceFiles/ncbiRefSeqCurated_NM.promotersUp2000Down1000AndExons.txt /data/hodges_lab/Tim/finalAnalyses_HMRs/HMRs_goodGenes/HMRs /data/hodges_lab/Tim/finalAnalyses_HMRs/HMRs_goodGenes/filtRefSeqFiles /data/hodges_lab/Tim/finalAnalyses_HMRs/HMRs_goodGenes/antiFiles\n",
    "done\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "07ed99b4",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Variables:\n",
    "# expects 5 cell type BED files as input\n",
    "CT=$1\n",
    "CTFile=$2\n",
    "OthFile=$3 # e.g. Anti_Bcell.bed; a collection of regions from all other CTs\n",
    "TSSExonRef=$4 # i.e. blacklist file for merging boundaries in establishing clusters\n",
    "outDir=$5\n",
    "inDir=$6\n",
    "antiDir=$7\n",
    "\n",
    "\n",
    "# Echo the variable assignments for good measure\n",
    "echo \"Input Directory identified: ${inDir}\"\n",
    "echo \"Output Directory identified: ${outDir}\"\n",
    "echo \"Cell-type identified: ${CT}\"\n",
    "echo \"Cell-type file identified: ${CTFile}\"\n",
    "echo \"Other cell-type file identified: ${OthFile}\"\n",
    "echo \"Clustering Regions Whitelist identified: ${invFile}\"\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "################################################################################\n",
    "# Find ALL HMRs\n",
    "echo \"Finding all cell specific and shared HMRs\"\n",
    "\n",
    "##### ((0)) ALL_cellspecific\n",
    "##### Exclude HMRs overlapping with blacklist | Exclude HMRs from this celltype that overlap with those from other celltypes\n",
    "bedtools intersect -v -a $inDir/$CTFile -b $TSSExonRef | bedtools intersect -v -a - -b $antiDir/$OthFile > $outDir/$CT\\_all_cellspecific.txt\n",
    "echo \"Found cell-specific HMRs.\"\n",
    "\n",
    "##### ((00)) ALL_shared\n",
    "##### Exclude HMRs overlapping with blacklist | Include HMRs from this celltype that overlap with those from other celltypes\n",
    "bedtools intersect -v -a $inDir/$CTFile -b $TSSExonRef | bedtools intersect -u -a - -b $antiDir/$OthFile > $outDir/$CT\\_all_shared.txt\n",
    "echo \"Found shared HMRs.\"\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "################################################################################\n",
    "# Find Unclustered HMRs\n",
    "\n",
    "##### ((1)) CT_unclustered\n",
    "#####  Filt for TSS/Exons | Merge together clustered HMRs based on rules | Filter out unclustered\n",
    "originalHMRDir=/data/hodges_lab/Tim/finalAnalyses_HMRs/HMRs_goodGenes/startingBEDs/\n",
    "startingFile=${originalHMRDir}${CT}.hmr.bed\n",
    "\n",
    "awk 'BEGIN{OFS=FS=\"\\t\"}{if (($3-$2)>49) print $1,$2,$3}' ${startingFile} | bedtools merge -c 2 -o count -d 6000 -i - | awk 'BEGIN{FS=OFS=\"\\t\"}{if ($4<2) print}' - | awk 'BEGIN{OFS=FS=\"\\t\"}{print $1,$2,$3}' - | bedtools intersect -v -a - -b ${TSSExonRef} > $outDir/$CT\\_unclustered.txt\n",
    "echo \"Found unclustered HMRs.\"\n",
    "\n",
    "\n",
    "##### ((2)) CT_unclustered_cellspecific\n",
    "##### \n",
    "# Inverse intersect (1) with comparison file to find (a) ct-specific unclustered HMRs\n",
    "bedtools intersect -v -a $outDir/$CT\\_unclustered.txt -b $antiDir/$OthFile | awk 'BEGIN{OFS=FS=\"\\t\"}{print $1,$2,$3}' - > $outDir/$CT\\_unclustered_cellspecific.txt\n",
    "echo \"Found cell-specific unclustered HMRs.\"\n",
    "\n",
    "\n",
    "##### ((3) CT_unclustered_shared\n",
    "##### \n",
    "# Intersect (1) with comparison file to find (b) ct-shared unclustered HMRs\n",
    "bedtools intersect -u -a $outDir/$CT\\_unclustered.txt -b $antiDir/$OthFile | awk 'BEGIN{OFS=FS=\"\\t\"}{print $1,$2,$3}' - > $outDir/$CT\\_unclustered_shared.txt\n",
    "echo \"Found shared unclustered HMRs.\"\n",
    "\n",
    "\n",
    "################################################################################\n",
    "# Find Clustered HMRs\n",
    "\n",
    "##### ((4)) CT_internalClusters\n",
    "##### \n",
    "# internal Clusters that don't cross at TSS/Exon, that also compose 3+ HMRs\n",
    "# Strategy: \n",
    "#\t\t- (pre-a) Create a whitelist from the TSS/Exon blacklist\n",
    "# \t\t- (a) Find all inverse regions (Whitelist regions) that contain 3+ HMRs (potential clusters)\n",
    "# \t\t- (b) Make BED that has two BED coordinates per line - Left: Whitelist Region; Right: HMR region\n",
    "#\t\t- (c) Merge using this altered BED-BED file and a custom awk script instead of BEDtools Merge\n",
    "\n",
    "# (pre-a) Turn blacklist into whitelist\n",
    "# To download the hg19 chr size file: \n",
    "# mysql --user=genome --host=genome-mysql.cse.ucsc.edu -A -e \"select chrom, size from hg19.chromInfo\"  > hg19.genome\n",
    "# awk 'BEGIN{OFS=FS=\"\\t\"}{print $1,$2,$2}' /scratch/scottt7/paper_wases/HMRs/hg19.genome | bedtools sort -i - > /scratch/scottt7/paper_wases/HMRs/hg19.genome_sorted.txt\n",
    "awk 'BEGIN{OFS=FS=\"\\t\"}{if ($1 !~ /\\_/)print $1,$2,$3}' $TSSExonRef | bedtools sort -i - | bedtools complement -i - -g /scratch/scottt7/paper_wases/HMRs/hg19.chrom.sizes.twoCol.sorted > $outDir/$CT\\_whitelist.txt\n",
    "echo \"Made whitelist.\"\n",
    "\n",
    "\n",
    "# (a) Find whitelist regions with 3+ HMRs\n",
    "bedtools intersect -c -F 1.0 -a $outDir/$CT\\_whitelist.txt -b $inDir/$CTFile | awk 'BEGIN{OFS=FS=\"\\t\"}{if ($4>2) print}' - > $outDir/$CT\\_whitelistRegions_contain3ormoreHMRs.txt\n",
    "echo \"Found whitelist regions that contain 3+ HMRs.\"\n",
    "\n",
    "# (b)Find what HMRs are in these to create a double BED file in a parsable list for custom merging: [1-3]: BED coordinates of Whitelist region [4]: #HMR in Whitelist region, [5-7]: BED coordinates of HMRs in Whitelist region\n",
    "# Diagnostic: bedtools intersect -loj -F 1.0 -a Bcell_whitelistRegions_contain3ormoreHMRs.txt -b Bcell.HMR.min50.txt | awk 'BEGIN{OFS=FS=\"\\t\"}{print}' - | bedtools sort -i - | head\n",
    "# awk 'BEGIN{OFS=FS=\"\\t\"}{print $5,$6,$7,$1,$2,$3,$4}' temp.txt | head\n",
    "\n",
    "bedtools intersect -loj -F 1.0 -a $outDir/$CT\\_whitelistRegions_contain3ormoreHMRs.txt -b $inDir/$CTFile | awk 'BEGIN{OFS=FS=\"\\t\"}{print $5,$6,$7,$1,$2,$3,$4}' - | bedtools sort -i - > $outDir/$CT\\_whitelistRegions_cluster_individualHMRs_temp.txt\n",
    "echo \"Created double BED for whitelist regions that contain 3+ HMRs.\"\n",
    "\n",
    "# (c) Custom merge within these regions\n",
    "awk -v whitelistStart=1 -v hmrsCount=1 -v clusterChr=1 -v clusterStart=1 -v clusterEnd=2 'BEGIN{OFS=FS=\"\\t\";dist=6000;whitelistStart=1;hmrsCount=1;clusterChr=1;clusterStart=1;clusterEnd=2;} {\n",
    "\t# Check if we are in the same WhitelistBoundary\n",
    "\tif ($5!=whitelistStart) {\n",
    "\t\tprint clusterChr,clusterStart,clusterEnd,hmrsCount;\n",
    "\t\twhitelistStart=$5;\n",
    "\t\thmrsCount=1;\n",
    "\t\tclusterChr=$1;\n",
    "\t\tclusterStart=$2;\n",
    "\t\tclusterEnd=$3;\n",
    "\t} else {\n",
    "\t\t# case if (1) we are staying within a Whitelist boundary, but (2) the previous HMR was >6000bp away\n",
    "\t\tif (($2-clusterEnd)>6000) {\n",
    "\t\t\tprint clusterChr,clusterStart,clusterEnd,hmrsCount;\n",
    "\t\t\tclusterStart=$2;\n",
    "\t\t\tclusterEnd=$3;\n",
    "\t\t\thmrsCount=1;\n",
    "\t\t} else {\n",
    "\t\t\tclusterEnd=$3;\n",
    "\t\t\thmrsCount+=1;\n",
    "\t\t}\n",
    "\t}\n",
    "}' $outDir/$CT\\_whitelistRegions_cluster_individualHMRs_temp.txt | awk 'BEGIN{OFS=FS=\"\\t\"}{if ($4>2) print $0}' - > $outDir/$CT\\_internalClusters.txt\n",
    "echo \"Found internalClusters containing 3+ HMRs.\"\n",
    "\n",
    "#########################\n",
    "#########################\n",
    "#########################\n",
    "# Remove unplaced contigs, causes issues downline \n",
    "awk 'BEGIN{OFS=FS=\"\\t\"}{if ($1 !~ /\\_/)print $1,$2,$3}' $outDir/$CT\\_internalClusters.txt > $outDir/$CT\\_internalClusters_BED.txt\n",
    "echo $\"\\n\"\n",
    "\n",
    "\n",
    "##### ((5)) internalClusters_individualHMRs \n",
    "#####\n",
    "bedtools intersect -u -a $inDir/$CTFile -b $outDir/$CT\\_internalClusters_BED.txt > $outDir/$CT\\_internalClusters_individualHMRs.txt\n",
    "echo \"Found the HMRs composing internalClusters containing 3+ HMRs.\"\n",
    "\n",
    "##### ((6)) internalClusters_individualHMRs_cellspecific\n",
    "##### \n",
    "# Inverse intersect (1) with our comparison file to find (a) ct-specific and (b) at least partially shared regions\n",
    "bedtools intersect -v -a $outDir/$CT\\_internalClusters_individualHMRs.txt -b $antiDir/$OthFile > $outDir/$CT\\_internalClusters_individualHMRs_cellspecific.txt\n",
    "echo \"Found cell-specific clustered HMRs.\"\n",
    "\n",
    "\n",
    "##### ((7) CT_internalClusters_individualHMRs_shared\n",
    "##### \n",
    "#\tIntersect CT with OtherFile to find shared\n",
    "bedtools intersect -u -a $outDir/$CT\\_internalClusters_individualHMRs.txt -b $antiDir/$OthFile > $outDir/$CT\\_internalClusters_individualHMRs_shared.txt\n",
    "echo \"Found shared clustered HMRs.\"\n",
    "\n",
    "\n",
    "##### ((8)) internalClusters_containsCS\n",
    "#####\n",
    "bedtools intersect -u -a $outDir/$CT\\_internalClusters_BED.txt -b $outDir/$CT\\_all_cellspecific.txt > $outDir/$CT\\_internalClusters_containsCS.txt\n",
    "echo \"Found the HMRs composing internalClusters containing 3+ HMRs.\"\n",
    "\n",
    "\n",
    "##### ((9)) internalClusters_withoutCS \n",
    "#####\n",
    "bedtools intersect -v -a $outDir/$CT\\_internalClusters_BED.txt -b $outDir/$CT\\_all_cellspecific.txt > $outDir/$CT\\_internalClusters_withoutCS.txt\n",
    "echo \"Found the HMRs composing internalClusters containing 3+ HMRs.\""
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4ba0dc6f",
   "metadata": {},
   "source": [
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "42d58595",
   "metadata": {},
   "source": [
    "### Modified script - 11 kb"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "6157a0ad",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Got the variables! Thanks.\n",
      "Bcell\n",
      "/data/hodges_lab/Tim/finalAnalyses_HMRs/linkingDists/output/\n",
      "\n",
      "\n",
      "\n",
      "Found internalClusters containing 3+ HMRs.\n",
      "\\n\n",
      "Found the HMRs composing internalClusters containing 3+ HMRs.\n"
     ]
    }
   ],
   "source": [
    "# Variables:\n",
    "# The original script took in 7 variables which have been omitted here. \n",
    "# These described both input folders and files as well as other referenced files unnecessary\n",
    "#     for this step. \n",
    "\n",
    "CT=\"Bcell\"\n",
    "\n",
    "OUT_DIR=/data/hodges_lab/Tim/finalAnalyses_HMRs/linkingDists/output/\n",
    "\n",
    "# update terminal \n",
    "echo \"Got the variables! Thanks.\"\n",
    "echo ${CT}\n",
    "echo ${OUT_DIR}\n",
    "cd ${OUT_DIR}\n",
    "echo \n",
    "echo\n",
    "echo\n",
    "\n",
    "# (c) Custom merge within these regions\n",
    "awk -v whitelistStart=1 -v hmrsCount=1 -v clusterChr=1 -v clusterStart=1 -v clusterEnd=2 \\\n",
    "'BEGIN{OFS=FS=\"\\t\"; dist=11000; whitelistStart=1; hmrsCount=1; clusterChr=1; clusterStart=1; clusterEnd=2} \\\n",
    "{ if ($5!=whitelistStart) { \\\n",
    "print clusterChr,clusterStart,clusterEnd,hmrsCount;\\\n",
    "whitelistStart=$5;\\\n",
    "hmrsCount=1;\\\n",
    "clusterChr=$1;\\\n",
    "clusterStart=$2;\\\n",
    "clusterEnd=$3;\\\n",
    "} else {\\\n",
    "if (($2-clusterEnd)>dist) {\\\n",
    "print clusterChr,clusterStart,clusterEnd,hmrsCount;\\\n",
    "clusterStart=$2;\\\n",
    "clusterEnd=$3;\\\n",
    "hmrsCount=1;\\\n",
    "} else {\\\n",
    "clusterEnd=$3;\\\n",
    "hmrsCount+=1}}}' /data/hodges_lab/Tim/finalAnalyses_HMRs/hp_lineage/HMRs/Bcell_whitelistRegions_cluster_individualHMRs_temp_hpl.txt | \\\n",
    "awk 'BEGIN{OFS=FS=\"\\t\"}{if ($4>2) print $0}' - > ${OUT_DIR}Bcell_internalClusters_11000.txt\n",
    "\n",
    "echo \"Found internalClusters containing 3+ HMRs.\"\n",
    "\n",
    "#########################\n",
    "#########################\n",
    "#########################\n",
    "# Remove unplaced contigs, causes issues downline \n",
    "awk 'BEGIN{OFS=FS=\"\\t\"}{if ($1 !~ /\\_/)print $1,$2,$3}' ${OUT_DIR}Bcell_internalClusters_11000.txt \\\n",
    "> ${OUT_DIR}Bcell_internalClusters_BED_11000.txt\n",
    "echo $\"\\n\"\n",
    "\n",
    "\n",
    "##### ((5)) internalClusters_individualHMRs \n",
    "#####\n",
    "bedtools intersect -u -a \\\n",
    "/data/hodges_lab/Tim/finalAnalyses_HMRs/hp_lineage/filtRefSeqFiles/Bcell.minsize50.filtforrefseqTSSexons.txt -b ${OUT_DIR}Bcell_internalClusters_BED_11000.txt > ${OUT_DIR}Bcell_internalClusters_individualHMRs_11000.txt\n",
    "echo \"Found the HMRs composing internalClusters containing 3+ HMRs.\"\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "c30067e3",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "total 6.4M\n",
      "-rw-r--r-- 1 scottt7 hodges_lab  56K Jan 21  2022 Bcell_internalClusters_10500.txt\n",
      "-rw-r--r-- 1 scottt7 hodges_lab  51K Jan 21  2022 Bcell_internalClusters_BED_10500.txt\n",
      "-rw-r--r-- 1 scottt7 hodges_lab 241K Jan 21  2022 Bcell_internalClusters_individualHMRs_10500.txt\n",
      "-rw-r--r-- 1 scottt7 hodges_lab  60K Jan 21  2022 Bcell_internalClusters_12500.txt\n",
      "-rw-r--r-- 1 scottt7 hodges_lab  55K Jan 21  2022 Bcell_internalClusters_BED_12500.txt\n",
      "-rw-r--r-- 1 scottt7 hodges_lab 266K Jan 21  2022 Bcell_internalClusters_individualHMRs_12500.txt\n",
      "-rw-r--r-- 1 scottt7 hodges_lab  37K Jan 21  2022 Bcell_internalClusters_BED.txt\n",
      "-rw-r--r-- 1 scottt7 hodges_lab 303K Jan 24  2022 Bcell_unclustered_10500.txt\n",
      "-rw-r--r-- 1 scottt7 hodges_lab 273K Jan 24  2022 Bcell_unclustered_12500.txt\n",
      "-rw-r--r-- 1 scottt7 hodges_lab 1.4M Jan 24  2022 ChromHMM_BcellUnclustered_10500.txt\n",
      "-rw-r--r-- 1 scottt7 hodges_lab 831K Jan 24  2022 ChromHMM_BcellClustered_10500.txt\n",
      "-rw-r--r-- 1 scottt7 hodges_lab  366 Jan 24  2022 ChromHMM_BcellUnclustered_counts_10500.txt\n",
      "-rw-r--r-- 1 scottt7 hodges_lab  366 Jan 24  2022 ChromHMM_BcellClustered_counts_10500.txt\n",
      "-rw-r--r-- 1 scottt7 hodges_lab 1.3M Jan 24  2022 ChromHMM_BcellUnclustered_12500.txt\n",
      "-rw-r--r-- 1 scottt7 hodges_lab 928K Jan 24  2022 ChromHMM_BcellClustered_12500.txt\n",
      "-rw-r--r-- 1 scottt7 hodges_lab  366 Jan 24  2022 ChromHMM_BcellUnclustered_counts_12500.txt\n",
      "-rw-r--r-- 1 scottt7 hodges_lab  366 Jan 24  2022 ChromHMM_BcellClustered_counts_12500.txt\n",
      "-rw-r--r-- 1 scottt7 hodges_lab  57K Jul 11 03:29 Bcell_internalClusters_11000.txt\n",
      "-rw-r--r-- 1 scottt7 hodges_lab  52K Jul 11 03:29 Bcell_internalClusters_BED_11000.txt\n",
      "-rw-r--r-- 1 scottt7 hodges_lab 247K Jul 11 03:29 Bcell_internalClusters_individualHMRs_11000.txt\n"
     ]
    }
   ],
   "source": [
    "ls -lhtr "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "20d4ab5c",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  8805 Bcell_internalClusters_individualHMRs_10500.txt\n",
      "  9044 Bcell_internalClusters_individualHMRs_11000.txt\n",
      "  9717 Bcell_internalClusters_individualHMRs_12500.txt\n",
      " 27566 total\n"
     ]
    }
   ],
   "source": [
    "wc -l Bcell_internalClusters_individualHMRs*"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c1bb3f4f",
   "metadata": {},
   "source": [
    "### Unclustered 11 kb"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "bc1f63a4",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Found unclustered HMRs.\n"
     ]
    }
   ],
   "source": [
    "################################################################################\n",
    "# Find Unclustered HMRs\n",
    "TSSExonRef=/data/hodges_lab/Tim/finalAnalyses_HMRs/hp_lineage/referenceFiles/ncbiRefSeqCurated_NM.promotersUp2000Down1000AndExons.txt\n",
    "CT=\"Bcell\"\n",
    "OUT_DIR=/data/hodges_lab/Tim/finalAnalyses_HMRs/linkingDists/output/\n",
    "\n",
    "##### ((1)) CT_unclustered\n",
    "#####  Filt for TSS/Exons | Merge together clustered HMRs based on rules | Filter out unclustered\n",
    "startingFile=/data/hodges_lab/Tim/finalAnalyses_HMRs/hp_lineage/startingBEDs/Bcell.hmr.bed\n",
    "\n",
    "awk 'BEGIN{OFS=FS=\"\\t\"}{if (($3-$2)>49) print $1,$2,$3}' ${startingFile} | \\\n",
    "bedtools merge -c 2 -o count -d 11000 -i - | awk 'BEGIN{FS=OFS=\"\\t\"}{if ($4<2) print}' - | \\\n",
    "awk 'BEGIN{OFS=FS=\"\\t\"}{print $1,$2,$3}' - | bedtools intersect -v -a - -b ${TSSExonRef} > \\\n",
    "${OUT_DIR}Bcell_unclustered_11000.txt\n",
    "\n",
    "echo \"Found unclustered HMRs.\"\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "48b39ed9",
   "metadata": {},
   "outputs": [],
   "source": [
    "wc -l *11000.txt"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2bbf0095",
   "metadata": {},
   "source": [
    "### Modified script - 10.5 kb\n",
    "\n",
    "Requires a few modifications\n",
    "\n",
    "(1) New output folder\n",
    "\n",
    "    - New output folder variable for quick reference later\n",
    "    \n",
    "    - /data/hodges_lab/Tim/finalAnalyses_HMRs/linkingDists/output\n",
    "    \n",
    "(2) New output file name for clarity"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "e537ab32",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Got the variables! Thanks.\n",
      "Bcell\n",
      "/data/hodges_lab/Tim/finalAnalyses_HMRs/linkingDists/output\n",
      "\n",
      "\n",
      "\n",
      "Found internalClusters containing 3+ HMRs.\n",
      "\\n\n",
      "Found the HMRs composing internalClusters containing 3+ HMRs.\n"
     ]
    }
   ],
   "source": [
    "# Variables:\n",
    "# The original script took in 7 variables which have been omitted here. \n",
    "# These described both input folders and files as well as other referenced files unnecessary\n",
    "#     for this step. \n",
    "\n",
    "CT=\"Bcell\"\n",
    "inDir=/data/hodges_lab/Tim/finalAnalyses_HMRs/linkingDists/HMRs_6/HMRs\n",
    "outDir=/data/hodges_lab/Tim/finalAnalyses_HMRs/linkingDists/output\n",
    "\n",
    "# update terminal \n",
    "echo \"Got the variables! Thanks.\"\n",
    "echo ${CT}\n",
    "echo ${outDir}\n",
    "cd ${outDir}\n",
    "echo \n",
    "echo\n",
    "echo\n",
    "\n",
    "# (c) Custom merge within these regions\n",
    "awk -v whitelistStart=1 -v hmrsCount=1 -v clusterChr=1 -v clusterStart=1 -v clusterEnd=2 \\\n",
    "'BEGIN{OFS=FS=\"\\t\"; dist=10500; whitelistStart=1; hmrsCount=1; clusterChr=1; clusterStart=1; clusterEnd=2} \\\n",
    "{ if ($5!=whitelistStart) { \\\n",
    "print clusterChr,clusterStart,clusterEnd,hmrsCount;\\\n",
    "whitelistStart=$5;\\\n",
    "hmrsCount=1;\\\n",
    "clusterChr=$1;\\\n",
    "clusterStart=$2;\\\n",
    "clusterEnd=$3;\\\n",
    "} else {\\\n",
    "if (($2-clusterEnd)>dist) {\\\n",
    "print clusterChr,clusterStart,clusterEnd,hmrsCount;\\\n",
    "clusterStart=$2;\\\n",
    "clusterEnd=$3;\\\n",
    "hmrsCount=1;\\\n",
    "} else {\\\n",
    "clusterEnd=$3;\\\n",
    "hmrsCount+=1}}}' ${inDir}/$CT\\_whitelistRegions_cluster_individualHMRs_temp.txt | \\\n",
    "awk 'BEGIN{OFS=FS=\"\\t\"}{if ($4>2) print $0}' - > ${outDir}/$CT\\_internalClusters_10500.txt\n",
    "\n",
    "echo \"Found internalClusters containing 3+ HMRs.\"\n",
    "\n",
    "#########################\n",
    "#########################\n",
    "#########################\n",
    "# Remove unplaced contigs, causes issues downline \n",
    "awk 'BEGIN{OFS=FS=\"\\t\"}{if ($1 !~ /\\_/)print $1,$2,$3}' ${outDir}/$CT\\_internalClusters_10500.txt \\\n",
    "> ${outDir}/$CT\\_internalClusters_BED_10500.txt\n",
    "echo $\"\\n\"\n",
    "\n",
    "\n",
    "##### ((5)) internalClusters_individualHMRs \n",
    "#####\n",
    "bedtools intersect -u -a \\\n",
    "/data/hodges_lab/Tim/finalAnalyses_HMRs/linkingDists/HMRs_6/filtRefSeqFiles/Bcell.minsize50.filtforrefseqTSSexons.txt \\\n",
    "-b ${outDir}/$CT\\_internalClusters_BED_10500.txt > ${outDir}/$CT\\_internalClusters_individualHMRs_10500.txt\n",
    "echo \"Found the HMRs composing internalClusters containing 3+ HMRs.\"\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "49a783d9",
   "metadata": {},
   "source": [
    "### Modified script - 12.5kb\n",
    "\n",
    "Requires a few modifications\n",
    "(1) New output folder\n",
    "    - New output folder variable for quick reference later\n",
    "    - /data/hodges_lab/Tim/finalAnalyses_HMRs/linkingDists/output\n",
    "(2) New output file name for clarity"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "871f8aa1",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Got the variables! Thanks.\n",
      "Bcell\n",
      "/data/hodges_lab/Tim/finalAnalyses_HMRs/linkingDists/output\n",
      "\n",
      "\n",
      "\n",
      "Found internalClusters containing 3+ HMRs.\n",
      "\\n\n",
      "Found the HMRs composing internalClusters containing 3+ HMRs.\n"
     ]
    }
   ],
   "source": [
    "# Variables:\n",
    "# The original script took in 7 variables which have been omitted here. \n",
    "# These described both input folders and files as well as other referenced files unnecessary\n",
    "#     for this step. \n",
    "\n",
    "CT=\"Bcell\"\n",
    "inDir=/data/hodges_lab/Tim/finalAnalyses_HMRs/linkingDists/HMRs_6/HMRs\n",
    "outDir=/data/hodges_lab/Tim/finalAnalyses_HMRs/linkingDists/output\n",
    "\n",
    "# update terminal \n",
    "echo \"Got the variables! Thanks.\"\n",
    "echo ${CT}\n",
    "echo ${outDir}\n",
    "cd ${outDir}\n",
    "echo \n",
    "echo\n",
    "echo\n",
    "\n",
    "# (c) Custom merge within these regions\n",
    "awk -v whitelistStart=1 -v hmrsCount=1 -v clusterChr=1 -v clusterStart=1 -v clusterEnd=2 \\\n",
    "'BEGIN{OFS=FS=\"\\t\"; dist=12500; whitelistStart=1; hmrsCount=1; clusterChr=1; clusterStart=1; clusterEnd=2} \\\n",
    "{ if ($5!=whitelistStart) { \\\n",
    "print clusterChr,clusterStart,clusterEnd,hmrsCount;\\\n",
    "whitelistStart=$5;\\\n",
    "hmrsCount=1;\\\n",
    "clusterChr=$1;\\\n",
    "clusterStart=$2;\\\n",
    "clusterEnd=$3;\\\n",
    "} else {\\\n",
    "if (($2-clusterEnd)>dist) {\\\n",
    "print clusterChr,clusterStart,clusterEnd,hmrsCount;\\\n",
    "clusterStart=$2;\\\n",
    "clusterEnd=$3;\\\n",
    "hmrsCount=1;\\\n",
    "} else {\\\n",
    "clusterEnd=$3;\\\n",
    "hmrsCount+=1}}}' ${inDir}/$CT\\_whitelistRegions_cluster_individualHMRs_temp.txt | \\\n",
    "awk 'BEGIN{OFS=FS=\"\\t\"}{if ($4>2) print $0}' - > ${outDir}/$CT\\_internalClusters_12500.txt\n",
    "\n",
    "echo \"Found internalClusters containing 3+ HMRs.\"\n",
    "\n",
    "#########################\n",
    "#########################\n",
    "#########################\n",
    "# Remove unplaced contigs, causes issues downline \n",
    "awk 'BEGIN{OFS=FS=\"\\t\"}{if ($1 !~ /\\_/)print $1,$2,$3}' ${outDir}/$CT\\_internalClusters_12500.txt \\\n",
    "> ${outDir}/$CT\\_internalClusters_BED_12500.txt\n",
    "echo $\"\\n\"\n",
    "\n",
    "\n",
    "##### ((5)) internalClusters_individualHMRs \n",
    "#####\n",
    "bedtools intersect -u -a \\\n",
    "/data/hodges_lab/Tim/finalAnalyses_HMRs/linkingDists/HMRs_6/filtRefSeqFiles/Bcell.minsize50.filtforrefseqTSSexons.txt \\\n",
    "-b ${outDir}/$CT\\_internalClusters_BED_12500.txt > ${outDir}/$CT\\_internalClusters_individualHMRs_12500.txt\n",
    "echo \"Found the HMRs composing internalClusters containing 3+ HMRs.\""
   ]
  },
  {
   "cell_type": "markdown",
   "id": "65bd0bf9",
   "metadata": {},
   "source": [
    "### Move the 6-kb file over to output for easier processing"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "2c2f9a50",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Did it. Moved 6kb.\n"
     ]
    }
   ],
   "source": [
    "cp /data/hodges_lab/Tim/finalAnalyses_HMRs/linkingDists/HMRs_6/HMRs/Bcell_internalClusters_BED.txt \\\n",
    "/gpfs52/data/hodges_lab/Tim/finalAnalyses_HMRs/linkingDists/output\n",
    "\n",
    "echo \"Did it. Moved 6kb.\""
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6b23cd5a",
   "metadata": {},
   "source": [
    "### Unclustered 10.5kb"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "fc40dd08",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Found unclustered HMRs.\n"
     ]
    }
   ],
   "source": [
    "################################################################################\n",
    "# Find Unclustered HMRs\n",
    "TSSExonRef=/data/hodges_lab/Tim/finalAnalyses_HMRs/HMRs_goodGenes/referenceFiles/ncbiRefSeqCurated_NM.promotersUp2000Down1000AndExons.txt\n",
    "CT=\"Bcell\"\n",
    "outDir=/data/hodges_lab/Tim/finalAnalyses_HMRs/linkingDists/output\n",
    "\n",
    "##### ((1)) CT_unclustered\n",
    "#####  Filt for TSS/Exons | Merge together clustered HMRs based on rules | Filter out unclustered\n",
    "originalHMRDir=/data/hodges_lab/Tim/finalAnalyses_HMRs/HMRs_goodGenes/startingBEDs/\n",
    "startingFile=${originalHMRDir}${CT}.hmr.bed\n",
    "\n",
    "awk 'BEGIN{OFS=FS=\"\\t\"}{if (($3-$2)>49) print $1,$2,$3}' ${startingFile} | \\\n",
    "bedtools merge -c 2 -o count -d 10500 -i - | awk 'BEGIN{FS=OFS=\"\\t\"}{if ($4<2) print}' - | \\\n",
    "awk 'BEGIN{OFS=FS=\"\\t\"}{print $1,$2,$3}' - | bedtools intersect -v -a - -b ${TSSExonRef} > \\\n",
    "$outDir/$CT\\_unclustered_10500.txt\n",
    "\n",
    "echo \"Found unclustered HMRs.\"\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "56dba7ad",
   "metadata": {},
   "source": [
    "### Unclustered 12.5kb"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "e4cc2864",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Found unclustered HMRs.\n"
     ]
    }
   ],
   "source": [
    "################################################################################\n",
    "# Find Unclustered HMRs\n",
    "TSSExonRef=/data/hodges_lab/Tim/finalAnalyses_HMRs/HMRs_goodGenes/referenceFiles/ncbiRefSeqCurated_NM.promotersUp2000Down1000AndExons.txt\n",
    "CT=\"Bcell\"\n",
    "outDir=/data/hodges_lab/Tim/finalAnalyses_HMRs/linkingDists/output\n",
    "\n",
    "##### ((1)) CT_unclustered\n",
    "#####  Filt for TSS/Exons | Merge together clustered HMRs based on rules | Filter out unclustered\n",
    "originalHMRDir=/data/hodges_lab/Tim/finalAnalyses_HMRs/HMRs_goodGenes/startingBEDs/\n",
    "startingFile=${originalHMRDir}${CT}.hmr.bed\n",
    "\n",
    "awk 'BEGIN{OFS=FS=\"\\t\"}{if (($3-$2)>49) print $1,$2,$3}' ${startingFile} | \\\n",
    "bedtools merge -c 2 -o count -d 12500 -i - | awk 'BEGIN{FS=OFS=\"\\t\"}{if ($4<2) print}' - | \\\n",
    "awk 'BEGIN{OFS=FS=\"\\t\"}{print $1,$2,$3}' - | bedtools intersect -v -a - -b ${TSSExonRef} > \\\n",
    "$outDir/$CT\\_unclustered_12500.txt\n",
    "\n",
    "echo \"Found unclustered HMRs.\"\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "fdc502ae",
   "metadata": {},
   "source": [
    "### HMR pipeline: For the pie-bar HMR-annotation plot (3D), we want to know if the relative ratios hold true for other linking distances\n",
    "\n",
    "To do this, we need to apply the entire pipeline for HMR annotation using the two alternate linking distances\n",
    "\n",
    "#### 10.5kb linking distance"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b52b0fc6",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Code for how this is currently run. \n",
    "# Here, instead, hand-set these variables\n",
    "\n",
    "for celltype in Adrenal Bcell fHeart fSpinal H1ESC Liver\n",
    "do\n",
    "\tsh annotate_HMRs.sh ${celltype} \\\n",
    "    ${celltype}.minsize50.filtforrefseqTSSexons.txt \\\n",
    "    anti_${celltype}.txt \\\n",
    "    /data/hodges_lab/Tim/finalAnalyses_HMRs/HMRs_goodGenes/referenceFiles/ncbiRefSeqCurated_NM.promotersUp2000Down1000AndExons.txt \\\n",
    "    /data/hodges_lab/Tim/finalAnalyses_HMRs/HMRs_goodGenes/HMRs \\\n",
    "    /data/hodges_lab/Tim/finalAnalyses_HMRs/HMRs_goodGenes/filtRefSeqFiles \\\n",
    "    /data/hodges_lab/Tim/finalAnalyses_HMRs/HMRs_goodGenes/antiFiles\n",
    "done\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "c1b67fb2",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Input Directory identified: /data/hodges_lab/Tim/finalAnalyses_HMRs/HMRs_goodGenes/filtRefSeqFiles\n",
      "Output Directory identified: /data/hodges_lab/Tim/finalAnalyses_HMRs/linkingDists/HMRs_10500\n",
      "Cell-type identified: Bcell\n",
      "Cell-type file identified: Bcell.minsize50.filtforrefseqTSSexons.txt\n",
      "Other cell-type file identified: anti_Bcell.txt\n",
      "Clustering Regions Whitelist identified: \n",
      "Finding all cell specific and shared HMRs\n",
      "Found cell-specific HMRs.\n",
      "Found shared HMRs.\n",
      "Found unclustered HMRs.\n",
      "Found cell-specific unclustered HMRs.\n",
      "Found shared unclustered HMRs.\n",
      "\n",
      "a1.txt\n",
      "awk\n",
      "b1.txt\n",
      "Bcell_internalClusters_individualHMRs.txt\n",
      "B cell & Liver - Cluster Overlap Comparison.ipynb\n",
      "Bcell_Persistence_HSPC_revisement_CLUSTERED.ipynb\n",
      "Bcell_Persistence_HSPC_revisement-Copy1.ipynb\n",
      "Bcell_Persistence_HSPC_revisement.ipynb\n",
      "Bcell_unclustered.txt\n",
      "BroadHistone_Download_Log.ipynb\n",
      "ChromHMM_10500_12500_linkingDistances.ipynb\n",
      "count_Bcell_sh_H1Derived.txt\n",
      "count_Bcell_sh_HSPCDerived.txt\n",
      "count_HSPC_cs.txt\n",
      "count_HSPC_sh_HSPCDerived.txt\n",
      "count_HSPSC_sh_H1Derived.txt\n",
      "count_HSPSC_sh_HSPCDerived.txt\n",
      "count_Macrophage_sh_H1Derived.txt\n",
      "count_Macrophage_sh_HSPCDerived.txt\n",
      "eQTL_ML_Enrichment_Liver.ipynb\n",
      "eQTL_ML_Enrichment_Whole_Blood.ipynb\n",
      "H1ESC_immuneFocused_cellspecific.txt\n",
      "H1ESC_immuneFocused_clustered.txt\n",
      "H1ESC_immuneFocused_shared.txt\n",
      "H1ESC_immuneFocused.txt\n",
      "H1ESC_immuneFocused_unclustered.txt\n",
      "HMR_annotation.ipynb\n",
      "HSPCDerived_counts_clustered.txt\n",
      "HSPCDerived_counts_unclustered.txt\n",
      "HSPC_immuneFocused_cl_shared_H1Derived.txt\n",
      "HSPC_immuneFocused_cl_shared_HSPCDerived.txt\n",
      "HSPC_immuneFocused_un_shared_H1Derived.txt\n",
      "HSPC_immuneFocused_un_shared_HSPCDerived.txt\n",
      ".ipynb_checkpoints/\n",
      "LDSC_Liver_Developmental.ipynb\n",
      "LinkingDistances.ipynb\n",
      "Transcriptome Imputation by Hand.ipynb\n",
      "Untitled1.ipynb\n",
      "Untitled.ipynb\n",
      "weights_bcell_clustered.txt\n",
      "weights_bcell_unclustered.txt\n",
      "weights.bed\n",
      "weights_bed_noHeader.txt\n",
      "\n",
      "a1.txt\n",
      "awk\n",
      "b1.txt\n",
      "Bcell_internalClusters_individualHMRs.txt\n",
      "B cell & Liver - Cluster Overlap Comparison.ipynb\n",
      "Bcell_Persistence_HSPC_revisement_CLUSTERED.ipynb\n",
      "Bcell_Persistence_HSPC_revisement-Copy1.ipynb\n",
      "Bcell_Persistence_HSPC_revisement.ipynb\n",
      "Bcell_unclustered.txt\n",
      "BroadHistone_Download_Log.ipynb\n",
      "ChromHMM_10500_12500_linkingDistances.ipynb\n",
      "count_Bcell_sh_H1Derived.txt\n",
      "count_Bcell_sh_HSPCDerived.txt\n",
      "count_HSPC_cs.txt\n",
      "count_HSPC_sh_HSPCDerived.txt\n",
      "count_HSPSC_sh_H1Derived.txt\n",
      "count_HSPSC_sh_HSPCDerived.txt\n",
      "count_Macrophage_sh_H1Derived.txt\n",
      "count_Macrophage_sh_HSPCDerived.txt\n",
      "eQTL_ML_Enrichment_Liver.ipynb\n",
      "eQTL_ML_Enrichment_Whole_Blood.ipynb\n",
      "H1ESC_immuneFocused_cellspecific.txt\n",
      "H1ESC_immuneFocused_clustered.txt\n",
      "H1ESC_immuneFocused_shared.txt\n",
      "H1ESC_immuneFocused.txt\n",
      "H1ESC_immuneFocused_unclustered.txt\n",
      "HMR_annotation.ipynb\n",
      "HSPCDerived_counts_clustered.txt\n",
      "HSPCDerived_counts_unclustered.txt\n",
      "HSPC_immuneFocused_cl_shared_H1Derived.txt\n",
      "HSPC_immuneFocused_cl_shared_HSPCDerived.txt\n",
      "HSPC_immuneFocused_un_shared_H1Derived.txt\n",
      "HSPC_immuneFocused_un_shared_HSPCDerived.txt\n",
      ".ipynb_checkpoints/\n",
      "LDSC_Liver_Developmental.ipynb\n",
      "LinkingDistances.ipynb\n",
      "Transcriptome Imputation by Hand.ipynb\n",
      "Untitled1.ipynb\n",
      "Untitled.ipynb\n",
      "weights_bcell_clustered.txt\n",
      "weights_bcell_unclustered.txt\n",
      "weights.bed\n",
      "weights_bed_noHeader.txt\n",
      "Making whitelist.\n",
      "Made whitelist.\n",
      "Found whitelist regions that contain 3+ HMRs.\n",
      "Created double BED for whitelist regions that contain 3+ HMRs.\n",
      "chr1\t564470\t570301\t3\n",
      "chr1\t794016\t805500\t3\n",
      "chr1\t832618\t856881\t7\n",
      "chr1\t993725\t1006171\t6\n",
      "chr1\t1071770\t1080160\t3\n",
      "chr1\t1092645\t1106080\t5\n",
      "chr1\t1771819\t1779732\t3\n",
      "chr1\t1827360\t1841244\t6\n",
      "chr1\t2043760\t2065178\t8\n",
      "chr1\t2242977\t2247085\t3\n",
      "Found the internal clusters\n",
      "\\n\n",
      "Found the HMRs composing internalClusters containing 3+ HMRs.\n",
      "Found cell-specific clustered HMRs.\n",
      "Found shared clustered HMRs.\n",
      "Found the HMRs composing internalClusters containing 3+ HMRs that contain CS.\n",
      "Found the HMRs composing internalClusters containing 3+ HMRs that don't contain CS.\n"
     ]
    }
   ],
   "source": [
    "# Variables:\n",
    "# expects 5 cell type BED files as input\n",
    "CT=\"Bcell\"\n",
    "CTFile=Bcell.minsize50.filtforrefseqTSSexons.txt\n",
    "OthFile=anti_Bcell.txt # e.g. anti_Bcell.bed; a collection of regions from all other CTs\n",
    "TSSExonRef=/data/hodges_lab/Tim/finalAnalyses_HMRs/HMRs_goodGenes/referenceFiles/ncbiRefSeqCurated_NM.promotersUp2000Down1000AndExons.txt # i.e. blacklist file for merging boundaries in establishing clusters\n",
    "outDir=/data/hodges_lab/Tim/finalAnalyses_HMRs/linkingDists/HMRs_10500\n",
    "inDir=/data/hodges_lab/Tim/finalAnalyses_HMRs/HMRs_goodGenes/filtRefSeqFiles\n",
    "antiDir=/data/hodges_lab/Tim/finalAnalyses_HMRs/HMRs_goodGenes/antiFiles\n",
    "\n",
    "# For here, I'm creating a new variable for linking distance for the sake of consistency \n",
    "LINKING_DISTANCE=10500\n",
    "\n",
    "\n",
    "# Echo the variable assignments for good measure\n",
    "echo \"Input Directory identified: ${inDir}\"\n",
    "echo \"Output Directory identified: ${outDir}\"\n",
    "echo \"Cell-type identified: ${CT}\"\n",
    "echo \"Cell-type file identified: ${CTFile}\"\n",
    "echo \"Other cell-type file identified: ${OthFile}\"\n",
    "echo \"Clustering Regions Whitelist identified: ${invFile}\"\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "################################################################################\n",
    "# Find ALL HMRs\n",
    "echo \"Finding all cell specific and shared HMRs\"\n",
    "\n",
    "##### ((0)) ALL_cellspecific\n",
    "##### Exclude HMRs overlapping with blacklist | Exclude HMRs from this celltype that overlap with those from other celltypes\n",
    "bedtools intersect -v -a $inDir/$CTFile -b $TSSExonRef | bedtools intersect -v -a - -b $antiDir/$OthFile > \\\n",
    "$outDir/$CT\\_all_cellspecific.txt\n",
    "echo \"Found cell-specific HMRs.\"\n",
    "\n",
    "##### ((00)) ALL_shared\n",
    "##### Exclude HMRs overlapping with blacklist | Include HMRs from this celltype that overlap with those from other celltypes\n",
    "bedtools intersect -v -a $inDir/$CTFile -b $TSSExonRef | bedtools intersect -u -a - -b $antiDir/$OthFile > \\\n",
    "$outDir/$CT\\_all_shared.txt\n",
    "echo \"Found shared HMRs.\"\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "################################################################################\n",
    "# Find Unclustered HMRs\n",
    "\n",
    "##### ((1)) CT_unclustered\n",
    "#####  Filt for TSS/Exons | Merge together clustered HMRs based on rules | Filter out unclustered\n",
    "originalHMRDir=/data/hodges_lab/Tim/finalAnalyses_HMRs/HMRs_goodGenes/startingBEDs/\n",
    "startingFile=${originalHMRDir}${CT}.hmr.bed\n",
    "\n",
    "awk 'BEGIN{OFS=FS=\"\\t\"}{if (($3-$2)>49) print $1,$2,$3}' ${startingFile} | \\\n",
    "bedtools merge -c 2 -o count -d ${LINKING_DISTANCE} -i - | awk 'BEGIN{FS=OFS=\"\\t\"}{if ($4<2) print}' - | \\\n",
    "awk 'BEGIN{OFS=FS=\"\\t\"}{print $1,$2,$3}' - | bedtools intersect -v -a - -b ${TSSExonRef} > \\\n",
    "$outDir/$CT\\_unclustered.txt\n",
    "\n",
    "echo \"Found unclustered HMRs.\"\n",
    "\n",
    "\n",
    "##### ((2)) CT_unclustered_cellspecific\n",
    "##### \n",
    "# Inverse intersect (1) with comparison file to find (a) ct-specific unclustered HMRs\n",
    "bedtools intersect -v -a $outDir/$CT\\_unclustered.txt -b $antiDir/$OthFile | \\\n",
    "awk 'BEGIN{OFS=FS=\"\\t\"}{print $1,$2,$3}' - > $outDir/$CT\\_unclustered_cellspecific.txt\n",
    "echo \"Found cell-specific unclustered HMRs.\"\n",
    "\n",
    "\n",
    "##### ((3) CT_unclustered_shared\n",
    "##### \n",
    "# Intersect (1) with comparison file to find (b) ct-shared unclustered HMRs\n",
    "bedtools intersect -u -a $outDir/$CT\\_unclustered.txt -b $antiDir/$OthFile | \\\n",
    "awk 'BEGIN{OFS=FS=\"\\t\"}{print $1,$2,$3}' - > $outDir/$CT\\_unclustered_shared.txt\n",
    "echo \"Found shared unclustered HMRs.\"\n",
    "\n",
    "\n",
    "################################################################################\n",
    "# Find Clustered HMRs\n",
    "\n",
    "##### ((4)) CT_internalClusters\n",
    "##### \n",
    "# internal Clusters that don't cross at TSS/Exon, that also compose 3+ HMRs\n",
    "# Strategy: \n",
    "#\t\t- (pre-a) Create a whitelist from the TSS/Exon blacklist\n",
    "# \t\t- (a) Find all inverse regions (Whitelist regions) that contain 3+ HMRs (potential clusters)\n",
    "# \t\t- (b) Make BED that has two BED coordinates per line - Left: Whitelist Region; Right: HMR region\n",
    "#\t\t- (c) Merge using this altered BED-BED file and a custom awk script instead of BEDtools Merge\n",
    "\n",
    "# (pre-a) Turn blacklist into whitelist\n",
    "# To download the hg19 chr size file: \n",
    "# mysql --user=genome --host=genome-mysql.cse.ucsc.edu -A -e \"select chrom, size from hg19.chromInfo\"  > hg19.genome\n",
    "# awk 'BEGIN{OFS=FS=\"\\t\"}{print $1,$2,$2}' /scratch/scottt7/paper_wases/HMRs/hg19.genome | bedtools sort -i - > /scratch/scottt7/paper_wases/HMRs/hg19.genome_sorted.txt\n",
    "echo \"Making whitelist.\"\n",
    "awk 'BEGIN{OFS=FS=\"\\t\"}{if ($1 !~ /\\_/)print $1,$2,$3}' $TSSExonRef | bedtools sort -i - | \\\n",
    "bedtools complement -i - -g /scratch/scottt7/paper_wases/HMRs/hg19.chrom.sizes.twoCol.sorted > \\\n",
    "$outDir/$CT\\_whitelist.txt\n",
    "echo \"Made whitelist.\"\n",
    "\n",
    "\n",
    "# (a) Find whitelist regions with 3+ HMRs\n",
    "bedtools intersect -c -F 1.0 -a $outDir/$CT\\_whitelist.txt -b $inDir/$CTFile | \\\n",
    "awk 'BEGIN{OFS=FS=\"\\t\"}{if ($4>2) print}' - > $outDir/$CT\\_whitelistRegions_contain3ormoreHMRs.txt\n",
    "echo \"Found whitelist regions that contain 3+ HMRs.\"\n",
    "\n",
    "# (b)Find what HMRs are in these to create a double BED file in a parsable list for custom merging: [1-3]: BED coordinates of Whitelist region [4]: #HMR in Whitelist region, [5-7]: BED coordinates of HMRs in Whitelist region\n",
    "# Diagnostic: bedtools intersect -loj -F 1.0 -a Bcell_whitelistRegions_contain3ormoreHMRs.txt -b Bcell.HMR.min50.txt | awk 'BEGIN{OFS=FS=\"\\t\"}{print}' - | bedtools sort -i - | head\n",
    "# awk 'BEGIN{OFS=FS=\"\\t\"}{print $5,$6,$7,$1,$2,$3,$4}' temp.txt | head\n",
    "\n",
    "bedtools intersect -loj -F 1.0 -a $outDir/$CT\\_whitelistRegions_contain3ormoreHMRs.txt -b $inDir/$CTFile | \\\n",
    "awk 'BEGIN{OFS=FS=\"\\t\"}{print $5,$6,$7,$1,$2,$3,$4}' - | bedtools sort -i - > \\\n",
    "$outDir/$CT\\_whitelistRegions_cluster_individualHMRs_temp.txt\n",
    "echo \"Created double BED for whitelist regions that contain 3+ HMRs.\"\n",
    "\n",
    "## (c) Custom merge within these regions\n",
    "awk -v LINKINGDIST=${LINKING_DISTANCE} -v whitelistStart=1 -v hmrsCount=1 -v clusterChr=1 -v clusterStart=1 -v clusterEnd=2 \\\n",
    "'BEGIN{OFS=FS=\"\\t\"; whitelistStart=1; hmrsCount=1; clusterChr=1; clusterStart=1; clusterEnd=2} \\\n",
    "{ if ($5!=whitelistStart) { \\\n",
    "print clusterChr,clusterStart,clusterEnd,hmrsCount;\\\n",
    "whitelistStart=$5;\\\n",
    "hmrsCount=1;\\\n",
    "clusterChr=$1;\\\n",
    "clusterStart=$2;\\\n",
    "clusterEnd=$3;\\\n",
    "} else {\\\n",
    "if (($2-clusterEnd)>LINKINGDIST) \\\n",
    "{print clusterChr,clusterStart,clusterEnd,hmrsCount;\\\n",
    "clusterStart=$2;\\\n",
    "clusterEnd=$3;\\\n",
    "hmrsCount=1;\\\n",
    "} else {\\\n",
    "clusterEnd=$3;\\\n",
    "hmrsCount+=1}}}' ${outDir}/$CT\\_whitelistRegions_cluster_individualHMRs_temp.txt | \\\n",
    "awk 'BEGIN{OFS=FS=\"\\t\"}{if ($4>2) print $0}' - > ${outDir}/$CT\\_internalClusters.txt\n",
    "head ${outDir}/$CT\\_internalClusters.txt\n",
    "echo \"Found the internal clusters\"\n",
    "\n",
    "#########################\n",
    "#########################\n",
    "#########################\n",
    "# Remove unplaced contigs, causes issues downline \n",
    "awk 'BEGIN{OFS=FS=\"\\t\"}{if ($1 !~ /\\_/)print $1,$2,$3}' $outDir/$CT\\_internalClusters.txt > $outDir/$CT\\_internalClusters_BED.txt\n",
    "echo $\"\\n\"\n",
    "\n",
    "\n",
    "##### ((5)) internalClusters_individualHMRs \n",
    "#####\n",
    "bedtools intersect -u -a $inDir/$CTFile -b $outDir/$CT\\_internalClusters_BED.txt > $outDir/$CT\\_internalClusters_individualHMRs.txt\n",
    "echo \"Found the HMRs composing internalClusters containing 3+ HMRs.\"\n",
    "\n",
    "##### ((6)) internalClusters_individualHMRs_cellspecific\n",
    "##### \n",
    "# Inverse intersect (1) with our comparison file to find (a) ct-specific and (b) at least partially shared regions\n",
    "bedtools intersect -v -a $outDir/$CT\\_internalClusters_individualHMRs.txt -b $antiDir/$OthFile > $outDir/$CT\\_internalClusters_individualHMRs_cellspecific.txt\n",
    "echo \"Found cell-specific clustered HMRs.\"\n",
    "\n",
    "\n",
    "##### ((7) CT_internalClusters_individualHMRs_shared\n",
    "##### \n",
    "#\tIntersect CT with OtherFile to find shared\n",
    "bedtools intersect -u -a $outDir/$CT\\_internalClusters_individualHMRs.txt -b $antiDir/$OthFile > $outDir/$CT\\_internalClusters_individualHMRs_shared.txt\n",
    "echo \"Found shared clustered HMRs.\"\n",
    "\n",
    "\n",
    "##### ((8)) internalClusters_containsCS\n",
    "#####\n",
    "bedtools intersect -u -a $outDir/$CT\\_internalClusters_BED.txt -b $outDir/$CT\\_all_cellspecific.txt > $outDir/$CT\\_internalClusters_containsCS.txt\n",
    "echo \"Found the HMRs composing internalClusters containing 3+ HMRs that contain CS.\"\n",
    "\n",
    "\n",
    "##### ((9)) internalClusters_withoutCS \n",
    "#####\n",
    "bedtools intersect -v -a $outDir/$CT\\_internalClusters_BED.txt -b $outDir/$CT\\_all_cellspecific.txt > $outDir/$CT\\_internalClusters_withoutCS.txt\n",
    "echo \"Found the HMRs composing internalClusters containing 3+ HMRs that don't contain CS.\""
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a4759637",
   "metadata": {},
   "source": [
    "#### 12.5kb linking distance "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "fa5fb240",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Input Directory identified: /data/hodges_lab/Tim/finalAnalyses_HMRs/HMRs_goodGenes/filtRefSeqFiles\n",
      "Output Directory identified: /data/hodges_lab/Tim/finalAnalyses_HMRs/linkingDists/HMRs_12500\n",
      "Cell-type identified: Bcell\n",
      "Cell-type file identified: Bcell.minsize50.filtforrefseqTSSexons.txt\n",
      "Other cell-type file identified: anti_Bcell.txt\n",
      "Clustering Regions Whitelist identified: \n",
      "Finding all cell specific and shared HMRs\n",
      "Found cell-specific HMRs.\n",
      "Found shared HMRs.\n",
      "Found unclustered HMRs.\n",
      "Found cell-specific unclustered HMRs.\n",
      "Found shared unclustered HMRs.\n",
      "\n",
      "a1.txt\n",
      "awk\n",
      "b1.txt\n",
      "Bcell_internalClusters_individualHMRs.txt\n",
      "B cell & Liver - Cluster Overlap Comparison.ipynb\n",
      "Bcell_Persistence_HSPC_revisement_CLUSTERED.ipynb\n",
      "Bcell_Persistence_HSPC_revisement-Copy1.ipynb\n",
      "Bcell_Persistence_HSPC_revisement.ipynb\n",
      "Bcell_unclustered.txt\n",
      "BroadHistone_Download_Log.ipynb\n",
      "ChromHMM_10500_12500_linkingDistances.ipynb\n",
      "count_Bcell_sh_H1Derived.txt\n",
      "count_Bcell_sh_HSPCDerived.txt\n",
      "count_HSPC_cs.txt\n",
      "count_HSPC_sh_HSPCDerived.txt\n",
      "count_HSPSC_sh_H1Derived.txt\n",
      "count_HSPSC_sh_HSPCDerived.txt\n",
      "count_Macrophage_sh_H1Derived.txt\n",
      "count_Macrophage_sh_HSPCDerived.txt\n",
      "eQTL_ML_Enrichment_Liver.ipynb\n",
      "eQTL_ML_Enrichment_Whole_Blood.ipynb\n",
      "H1ESC_immuneFocused_cellspecific.txt\n",
      "H1ESC_immuneFocused_clustered.txt\n",
      "H1ESC_immuneFocused_shared.txt\n",
      "H1ESC_immuneFocused.txt\n",
      "H1ESC_immuneFocused_unclustered.txt\n",
      "HMR_annotation.ipynb\n",
      "HSPCDerived_counts_clustered.txt\n",
      "HSPCDerived_counts_unclustered.txt\n",
      "HSPC_immuneFocused_cl_shared_H1Derived.txt\n",
      "HSPC_immuneFocused_cl_shared_HSPCDerived.txt\n",
      "HSPC_immuneFocused_un_shared_H1Derived.txt\n",
      "HSPC_immuneFocused_un_shared_HSPCDerived.txt\n",
      ".ipynb_checkpoints/\n",
      "LDSC_Liver_Developmental.ipynb\n",
      "LinkingDistances.ipynb\n",
      "Transcriptome Imputation by Hand.ipynb\n",
      "Untitled1.ipynb\n",
      "Untitled.ipynb\n",
      "weights_bcell_clustered.txt\n",
      "weights_bcell_unclustered.txt\n",
      "weights.bed\n",
      "weights_bed_noHeader.txt\n",
      "\n",
      "a1.txt\n",
      "awk\n",
      "b1.txt\n",
      "Bcell_internalClusters_individualHMRs.txt\n",
      "B cell & Liver - Cluster Overlap Comparison.ipynb\n",
      "Bcell_Persistence_HSPC_revisement_CLUSTERED.ipynb\n",
      "Bcell_Persistence_HSPC_revisement-Copy1.ipynb\n",
      "Bcell_Persistence_HSPC_revisement.ipynb\n",
      "Bcell_unclustered.txt\n",
      "BroadHistone_Download_Log.ipynb\n",
      "ChromHMM_10500_12500_linkingDistances.ipynb\n",
      "count_Bcell_sh_H1Derived.txt\n",
      "count_Bcell_sh_HSPCDerived.txt\n",
      "count_HSPC_cs.txt\n",
      "count_HSPC_sh_HSPCDerived.txt\n",
      "count_HSPSC_sh_H1Derived.txt\n",
      "count_HSPSC_sh_HSPCDerived.txt\n",
      "count_Macrophage_sh_H1Derived.txt\n",
      "count_Macrophage_sh_HSPCDerived.txt\n",
      "eQTL_ML_Enrichment_Liver.ipynb\n",
      "eQTL_ML_Enrichment_Whole_Blood.ipynb\n",
      "H1ESC_immuneFocused_cellspecific.txt\n",
      "H1ESC_immuneFocused_clustered.txt\n",
      "H1ESC_immuneFocused_shared.txt\n",
      "H1ESC_immuneFocused.txt\n",
      "H1ESC_immuneFocused_unclustered.txt\n",
      "HMR_annotation.ipynb\n",
      "HSPCDerived_counts_clustered.txt\n",
      "HSPCDerived_counts_unclustered.txt\n",
      "HSPC_immuneFocused_cl_shared_H1Derived.txt\n",
      "HSPC_immuneFocused_cl_shared_HSPCDerived.txt\n",
      "HSPC_immuneFocused_un_shared_H1Derived.txt\n",
      "HSPC_immuneFocused_un_shared_HSPCDerived.txt\n",
      ".ipynb_checkpoints/\n",
      "LDSC_Liver_Developmental.ipynb\n",
      "LinkingDistances.ipynb\n",
      "Transcriptome Imputation by Hand.ipynb\n",
      "Untitled1.ipynb\n",
      "Untitled.ipynb\n",
      "weights_bcell_clustered.txt\n",
      "weights_bcell_unclustered.txt\n",
      "weights.bed\n",
      "weights_bed_noHeader.txt\n",
      "Making whitelist.\n",
      "Made whitelist.\n",
      "Found whitelist regions that contain 3+ HMRs.\n",
      "Created double BED for whitelist regions that contain 3+ HMRs.\n",
      "chr1\t564470\t570301\t3\n",
      "chr1\t794016\t805500\t3\n",
      "chr1\t832618\t856881\t7\n",
      "chr1\t993725\t1006171\t6\n",
      "chr1\t1071770\t1106080\t8\n",
      "chr1\t1771819\t1779732\t3\n",
      "chr1\t1827360\t1841244\t6\n",
      "chr1\t2043760\t2065178\t8\n",
      "chr1\t2162505\t2178389\t3\n",
      "chr1\t2206661\t2233250\t4\n",
      "Found the internal clusters\n",
      "\\n\n",
      "Found the HMRs composing internalClusters containing 3+ HMRs.\n",
      "Found cell-specific clustered HMRs.\n",
      "Found shared clustered HMRs.\n",
      "Found the HMRs composing internalClusters containing 3+ HMRs that contain CS.\n",
      "Found the HMRs composing internalClusters containing 3+ HMRs that don't contain CS.\n"
     ]
    }
   ],
   "source": [
    "# Variables:\n",
    "# expects 5 cell type BED files as input\n",
    "CT=\"Bcell\"\n",
    "CTFile=Bcell.minsize50.filtforrefseqTSSexons.txt\n",
    "OthFile=anti_Bcell.txt # e.g. anti_Bcell.bed; a collection of regions from all other CTs\n",
    "TSSExonRef=/data/hodges_lab/Tim/finalAnalyses_HMRs/HMRs_goodGenes/referenceFiles/ncbiRefSeqCurated_NM.promotersUp2000Down1000AndExons.txt # i.e. blacklist file for merging boundaries in establishing clusters\n",
    "outDir=/data/hodges_lab/Tim/finalAnalyses_HMRs/linkingDists/HMRs_12500\n",
    "inDir=/data/hodges_lab/Tim/finalAnalyses_HMRs/HMRs_goodGenes/filtRefSeqFiles\n",
    "antiDir=/data/hodges_lab/Tim/finalAnalyses_HMRs/HMRs_goodGenes/antiFiles\n",
    "\n",
    "# For here, I'm creating a new variable for linking distance for the sake of consistency \n",
    "LINKING_DISTANCE=12500\n",
    "\n",
    "\n",
    "# Echo the variable assignments for good measure\n",
    "echo \"Input Directory identified: ${inDir}\"\n",
    "echo \"Output Directory identified: ${outDir}\"\n",
    "echo \"Cell-type identified: ${CT}\"\n",
    "echo \"Cell-type file identified: ${CTFile}\"\n",
    "echo \"Other cell-type file identified: ${OthFile}\"\n",
    "echo \"Clustering Regions Whitelist identified: ${invFile}\"\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "################################################################################\n",
    "# Find ALL HMRs\n",
    "echo \"Finding all cell specific and shared HMRs\"\n",
    "\n",
    "##### ((0)) ALL_cellspecific\n",
    "##### Exclude HMRs overlapping with blacklist | Exclude HMRs from this celltype that overlap with those from other celltypes\n",
    "bedtools intersect -v -a $inDir/$CTFile -b $TSSExonRef | bedtools intersect -v -a - -b $antiDir/$OthFile > \\\n",
    "$outDir/$CT\\_all_cellspecific.txt\n",
    "echo \"Found cell-specific HMRs.\"\n",
    "\n",
    "##### ((00)) ALL_shared\n",
    "##### Exclude HMRs overlapping with blacklist | Include HMRs from this celltype that overlap with those from other celltypes\n",
    "bedtools intersect -v -a $inDir/$CTFile -b $TSSExonRef | bedtools intersect -u -a - -b $antiDir/$OthFile > \\\n",
    "$outDir/$CT\\_all_shared.txt\n",
    "echo \"Found shared HMRs.\"\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "################################################################################\n",
    "# Find Unclustered HMRs\n",
    "\n",
    "##### ((1)) CT_unclustered\n",
    "#####  Filt for TSS/Exons | Merge together clustered HMRs based on rules | Filter out unclustered\n",
    "originalHMRDir=/data/hodges_lab/Tim/finalAnalyses_HMRs/HMRs_goodGenes/startingBEDs/\n",
    "startingFile=${originalHMRDir}${CT}.hmr.bed\n",
    "\n",
    "awk 'BEGIN{OFS=FS=\"\\t\"}{if (($3-$2)>49) print $1,$2,$3}' ${startingFile} | \\\n",
    "bedtools merge -c 2 -o count -d ${LINKING_DISTANCE} -i - | awk 'BEGIN{FS=OFS=\"\\t\"}{if ($4<2) print}' - | \\\n",
    "awk 'BEGIN{OFS=FS=\"\\t\"}{print $1,$2,$3}' - | bedtools intersect -v -a - -b ${TSSExonRef} > \\\n",
    "$outDir/$CT\\_unclustered.txt\n",
    "\n",
    "echo \"Found unclustered HMRs.\"\n",
    "\n",
    "\n",
    "##### ((2)) CT_unclustered_cellspecific\n",
    "##### \n",
    "# Inverse intersect (1) with comparison file to find (a) ct-specific unclustered HMRs\n",
    "bedtools intersect -v -a $outDir/$CT\\_unclustered.txt -b $antiDir/$OthFile | \\\n",
    "awk 'BEGIN{OFS=FS=\"\\t\"}{print $1,$2,$3}' - > $outDir/$CT\\_unclustered_cellspecific.txt\n",
    "echo \"Found cell-specific unclustered HMRs.\"\n",
    "\n",
    "\n",
    "##### ((3) CT_unclustered_shared\n",
    "##### \n",
    "# Intersect (1) with comparison file to find (b) ct-shared unclustered HMRs\n",
    "bedtools intersect -u -a $outDir/$CT\\_unclustered.txt -b $antiDir/$OthFile | \\\n",
    "awk 'BEGIN{OFS=FS=\"\\t\"}{print $1,$2,$3}' - > $outDir/$CT\\_unclustered_shared.txt\n",
    "echo \"Found shared unclustered HMRs.\"\n",
    "\n",
    "\n",
    "################################################################################\n",
    "# Find Clustered HMRs\n",
    "\n",
    "##### ((4)) CT_internalClusters\n",
    "##### \n",
    "# internal Clusters that don't cross at TSS/Exon, that also compose 3+ HMRs\n",
    "# Strategy: \n",
    "#\t\t- (pre-a) Create a whitelist from the TSS/Exon blacklist\n",
    "# \t\t- (a) Find all inverse regions (Whitelist regions) that contain 3+ HMRs (potential clusters)\n",
    "# \t\t- (b) Make BED that has two BED coordinates per line - Left: Whitelist Region; Right: HMR region\n",
    "#\t\t- (c) Merge using this altered BED-BED file and a custom awk script instead of BEDtools Merge\n",
    "\n",
    "# (pre-a) Turn blacklist into whitelist\n",
    "# To download the hg19 chr size file: \n",
    "# mysql --user=genome --host=genome-mysql.cse.ucsc.edu -A -e \"select chrom, size from hg19.chromInfo\"  > hg19.genome\n",
    "# awk 'BEGIN{OFS=FS=\"\\t\"}{print $1,$2,$2}' /scratch/scottt7/paper_wases/HMRs/hg19.genome | bedtools sort -i - > /scratch/scottt7/paper_wases/HMRs/hg19.genome_sorted.txt\n",
    "echo \"Making whitelist.\"\n",
    "awk 'BEGIN{OFS=FS=\"\\t\"}{if ($1 !~ /\\_/)print $1,$2,$3}' $TSSExonRef | bedtools sort -i - | \\\n",
    "bedtools complement -i - -g /scratch/scottt7/paper_wases/HMRs/hg19.chrom.sizes.twoCol.sorted > \\\n",
    "$outDir/$CT\\_whitelist.txt\n",
    "echo \"Made whitelist.\"\n",
    "\n",
    "\n",
    "# (a) Find whitelist regions with 3+ HMRs\n",
    "bedtools intersect -c -F 1.0 -a $outDir/$CT\\_whitelist.txt -b $inDir/$CTFile | \\\n",
    "awk 'BEGIN{OFS=FS=\"\\t\"}{if ($4>2) print}' - > $outDir/$CT\\_whitelistRegions_contain3ormoreHMRs.txt\n",
    "echo \"Found whitelist regions that contain 3+ HMRs.\"\n",
    "\n",
    "# (b)Find what HMRs are in these to create a double BED file in a parsable list for custom merging: [1-3]: BED coordinates of Whitelist region [4]: #HMR in Whitelist region, [5-7]: BED coordinates of HMRs in Whitelist region\n",
    "# Diagnostic: bedtools intersect -loj -F 1.0 -a Bcell_whitelistRegions_contain3ormoreHMRs.txt -b Bcell.HMR.min50.txt | awk 'BEGIN{OFS=FS=\"\\t\"}{print}' - | bedtools sort -i - | head\n",
    "# awk 'BEGIN{OFS=FS=\"\\t\"}{print $5,$6,$7,$1,$2,$3,$4}' temp.txt | head\n",
    "\n",
    "bedtools intersect -loj -F 1.0 -a $outDir/$CT\\_whitelistRegions_contain3ormoreHMRs.txt -b $inDir/$CTFile | \\\n",
    "awk 'BEGIN{OFS=FS=\"\\t\"}{print $5,$6,$7,$1,$2,$3,$4}' - | bedtools sort -i - > \\\n",
    "$outDir/$CT\\_whitelistRegions_cluster_individualHMRs_temp.txt\n",
    "echo \"Created double BED for whitelist regions that contain 3+ HMRs.\"\n",
    "\n",
    "## (c) Custom merge within these regions\n",
    "awk -v LINKINGDIST=${LINKING_DISTANCE} -v whitelistStart=1 -v hmrsCount=1 -v clusterChr=1 -v clusterStart=1 -v clusterEnd=2 \\\n",
    "'BEGIN{OFS=FS=\"\\t\"; whitelistStart=1; hmrsCount=1; clusterChr=1; clusterStart=1; clusterEnd=2} \\\n",
    "{ if ($5!=whitelistStart) { \\\n",
    "print clusterChr,clusterStart,clusterEnd,hmrsCount;\\\n",
    "whitelistStart=$5;\\\n",
    "hmrsCount=1;\\\n",
    "clusterChr=$1;\\\n",
    "clusterStart=$2;\\\n",
    "clusterEnd=$3;\\\n",
    "} else {\\\n",
    "if (($2-clusterEnd)>LINKINGDIST) \\\n",
    "{print clusterChr,clusterStart,clusterEnd,hmrsCount;\\\n",
    "clusterStart=$2;\\\n",
    "clusterEnd=$3;\\\n",
    "hmrsCount=1;\\\n",
    "} else {\\\n",
    "clusterEnd=$3;\\\n",
    "hmrsCount+=1}}}' ${outDir}/$CT\\_whitelistRegions_cluster_individualHMRs_temp.txt | \\\n",
    "awk 'BEGIN{OFS=FS=\"\\t\"}{if ($4>2) print $0}' - > ${outDir}/$CT\\_internalClusters.txt\n",
    "head ${outDir}/$CT\\_internalClusters.txt\n",
    "echo \"Found the internal clusters\"\n",
    "\n",
    "#########################\n",
    "#########################\n",
    "#########################\n",
    "# Remove unplaced contigs, causes issues downline \n",
    "awk 'BEGIN{OFS=FS=\"\\t\"}{if ($1 !~ /\\_/)print $1,$2,$3}' $outDir/$CT\\_internalClusters.txt > $outDir/$CT\\_internalClusters_BED.txt\n",
    "echo $\"\\n\"\n",
    "\n",
    "\n",
    "##### ((5)) internalClusters_individualHMRs \n",
    "#####\n",
    "bedtools intersect -u -a $inDir/$CTFile -b $outDir/$CT\\_internalClusters_BED.txt > $outDir/$CT\\_internalClusters_individualHMRs.txt\n",
    "echo \"Found the HMRs composing internalClusters containing 3+ HMRs.\"\n",
    "\n",
    "##### ((6)) internalClusters_individualHMRs_cellspecific\n",
    "##### \n",
    "# Inverse intersect (1) with our comparison file to find (a) ct-specific and (b) at least partially shared regions\n",
    "bedtools intersect -v -a $outDir/$CT\\_internalClusters_individualHMRs.txt -b $antiDir/$OthFile > $outDir/$CT\\_internalClusters_individualHMRs_cellspecific.txt\n",
    "echo \"Found cell-specific clustered HMRs.\"\n",
    "\n",
    "\n",
    "##### ((7) CT_internalClusters_individualHMRs_shared\n",
    "##### \n",
    "#\tIntersect CT with OtherFile to find shared\n",
    "bedtools intersect -u -a $outDir/$CT\\_internalClusters_individualHMRs.txt -b $antiDir/$OthFile > $outDir/$CT\\_internalClusters_individualHMRs_shared.txt\n",
    "echo \"Found shared clustered HMRs.\"\n",
    "\n",
    "\n",
    "##### ((8)) internalClusters_containsCS\n",
    "#####\n",
    "bedtools intersect -u -a $outDir/$CT\\_internalClusters_BED.txt -b $outDir/$CT\\_all_cellspecific.txt > $outDir/$CT\\_internalClusters_containsCS.txt\n",
    "echo \"Found the HMRs composing internalClusters containing 3+ HMRs that contain CS.\"\n",
    "\n",
    "\n",
    "##### ((9)) internalClusters_withoutCS \n",
    "#####\n",
    "bedtools intersect -v -a $outDir/$CT\\_internalClusters_BED.txt -b $outDir/$CT\\_all_cellspecific.txt > $outDir/$CT\\_internalClusters_withoutCS.txt\n",
    "echo \"Found the HMRs composing internalClusters containing 3+ HMRs that don't contain CS.\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fad5ed39",
   "metadata": {},
   "outputs": [],
   "source": [
    "echo \"Hello\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "98f3d078",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {},
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": false,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {
    "height": "calc(100% - 180px)",
    "left": "10px",
    "top": "150px",
    "width": "420px"
   },
   "toc_section_display": true,
   "toc_window_display": true
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
